{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e1d8c303",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Project root added to sys.path: /Users/thiloweber/Code/3r-monitoring\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/thiloweber/miniforge3/envs/3r-monitoring/lib/python3.11/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n",
      "/Users/thiloweber/miniforge3/envs/3r-monitoring/lib/python3.11/site-packages/torch/_utils.py:831: UserWarning: TypedStorage is deprecated. It will be removed in the future and UntypedStorage will be the only storage class. This should only matter to you if you are using storages directly.  To access UntypedStorage directly, use tensor.untyped_storage() instead of tensor.storage()\n",
      "  return self.fget.__get__(instance, owner)()\n",
      "All PyTorch model weights were used when initializing TFBertModel.\n",
      "\n",
      "All the weights of TFBertModel were initialized from the PyTorch model.\n",
      "If your task is similar to the task the model of the checkpoint was trained on, you can already use TFBertModel for predictions without further training.\n"
     ]
    }
   ],
   "source": [
    "import sys\n",
    "from pathlib import Path\n",
    "\n",
    "# Add the src directory to the system path\n",
    "project_root = Path().resolve().parent  # Adjust if necessary\n",
    "sys.path.append(str(project_root))\n",
    "print(f\"Project root added to sys.path: {project_root}\")\n",
    "\n",
    "# Import necessary modules\n",
    "from src.models import biobert_goldhamster as goldh\n",
    "\n",
    "# Setup reloading all changed modules every time before executing a cell\n",
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "050509e0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMID</th>\n",
       "      <th>in_silico</th>\n",
       "      <th>organs</th>\n",
       "      <th>other</th>\n",
       "      <th>human</th>\n",
       "      <th>in_vivo</th>\n",
       "      <th>invertebrate</th>\n",
       "      <th>primary_cells</th>\n",
       "      <th>immortal_cell_line</th>\n",
       "      <th>TEXT</th>\n",
       "      <th>in_silico_label</th>\n",
       "      <th>organs_label</th>\n",
       "      <th>other_label</th>\n",
       "      <th>human_label</th>\n",
       "      <th>in_vivo_label</th>\n",
       "      <th>invertebrate_label</th>\n",
       "      <th>primary_cells_label</th>\n",
       "      <th>immortal_cell_line_label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>31705587</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Neurodegeneration refers to the complex proces...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>30807815</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>The present study was aimed at broadening the ...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>30594027</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>The increase of opportunistic fungal infection...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>30879321</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>The phospholipase C (PLC) is a family of kinas...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>30818821</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Mycoplasma gallisepticum (MG) mainly infects c...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>30074186</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>This study was designed to evaluate the effect...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>31150628</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>The dose-dependent neuroprotective role of lic...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>31104401</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Both Plasmodium spp. and Toxoplasma gondii are...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>29390112</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Most animals display retarded growth in advers...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>29381289</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>We constructed a UC-VEGF-SMC three dimensional...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>71 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        PMID  in_silico  organs  other  human  in_vivo  invertebrate  \\\n",
       "0   31705587          0       0      0      0        0             0   \n",
       "1   30807815          0       0      0      1        1             0   \n",
       "2   30594027          0       0      0      1        0             1   \n",
       "3   30879321          0       0      0      0        0             0   \n",
       "4   30818821          0       0      0      0        0             0   \n",
       "..       ...        ...     ...    ...    ...      ...           ...   \n",
       "66  30074186          0       0      0      0        1             0   \n",
       "67  31150628          0       0      0      0        1             0   \n",
       "68  31104401          0       0      0      0        1             0   \n",
       "69  29390112          0       0      0      0        1             0   \n",
       "70  29381289          0       0      0      0        1             0   \n",
       "\n",
       "    primary_cells  immortal_cell_line  \\\n",
       "0               0                   0   \n",
       "1               1                   1   \n",
       "2               0                   1   \n",
       "3               0                   1   \n",
       "4               0                   1   \n",
       "..            ...                 ...   \n",
       "66              0                   0   \n",
       "67              0                   0   \n",
       "68              0                   0   \n",
       "69              0                   0   \n",
       "70              0                   0   \n",
       "\n",
       "                                                 TEXT in_silico_label  \\\n",
       "0   Neurodegeneration refers to the complex proces...               0   \n",
       "1   The present study was aimed at broadening the ...               0   \n",
       "2   The increase of opportunistic fungal infection...               0   \n",
       "3   The phospholipase C (PLC) is a family of kinas...               0   \n",
       "4   Mycoplasma gallisepticum (MG) mainly infects c...               0   \n",
       "..                                                ...             ...   \n",
       "66  This study was designed to evaluate the effect...               0   \n",
       "67  The dose-dependent neuroprotective role of lic...               0   \n",
       "68  Both Plasmodium spp. and Toxoplasma gondii are...               0   \n",
       "69  Most animals display retarded growth in advers...               0   \n",
       "70  We constructed a UC-VEGF-SMC three dimensional...               0   \n",
       "\n",
       "   organs_label other_label human_label in_vivo_label invertebrate_label  \\\n",
       "0             0           0           0             0                  0   \n",
       "1             0           0           1             1                  0   \n",
       "2             0           0           1             0                  1   \n",
       "3             0           0           0             0                  0   \n",
       "4             0           0           0             0                  0   \n",
       "..          ...         ...         ...           ...                ...   \n",
       "66            0           0           0             1                  0   \n",
       "67            0           0           0             1                  0   \n",
       "68            0           0           0             1                  0   \n",
       "69            0           0           0             1                  0   \n",
       "70            0           0           0             1                  0   \n",
       "\n",
       "   primary_cells_label immortal_cell_line_label  \n",
       "0                    0                        0  \n",
       "1                    1                        1  \n",
       "2                    0                        1  \n",
       "3                    0                        1  \n",
       "4                    0                        1  \n",
       "..                 ...                      ...  \n",
       "66                   0                        0  \n",
       "67                   0                        0  \n",
       "68                   0                        0  \n",
       "69                   0                        0  \n",
       "70                   0                        0  \n",
       "\n",
       "[71 rows x 18 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import json\n",
    "\n",
    "papers_dir = project_root / \"data\" / \"goldhamster\" / \"papers\"\n",
    "labels_dir = project_root / \"data\" / \"goldhamster\" / \"labels\"\n",
    "\n",
    "# labels\n",
    "all_labels = ['in_silico','organs','other','human','in_vivo','invertebrate','primary_cells','immortal_cell_line']\n",
    "\n",
    "#######################################\n",
    "### --------- Import text --------- ###\n",
    "def read_text(pmid, docs_dir):\n",
    "\t# file = os.path.join(docs_dir, pmid+\".txt\")\n",
    "\t# with open(txt_file, \"r\") as text_file:\n",
    "\t# \ttext = text_file.read()\n",
    "\t# \ttext = text.replace(\"\\n\",\" \")\n",
    "\t# \ttext = text.replace(\"\\t\",\" \")\n",
    "\n",
    "\tfile = docs_dir / f\"{pmid}.json\"\n",
    "\tpaper_dict = json.loads(file.read_text())\n",
    "\ttitle = paper_dict[\"title\"]\n",
    "\tabstract = paper_dict[\"abstract\"]\n",
    "\tmesh_terms = paper_dict[\"mesh_terms\"]\n",
    "\n",
    "\t# return f\"{title} {abstract}\".replace(\"\\n\",\" \").replace(\"\\t\",\" \")\n",
    "\treturn f\"{abstract}\"\n",
    "\n",
    "#######################################\n",
    "### -------- Import Splits -------- ###\n",
    "def import_splits(docs_dir,train_dev_test_dir,filename):\n",
    "\ttsv_file = filename[0:-5]+\".tsv\"\n",
    "\twith open(os.path.join(tsv_file), \"w\") as writer:\n",
    "\t\tline = \"PMID\"\n",
    "\t\tfor label in all_labels:\n",
    "\t\t\tline += \"\\t\"+label\n",
    "\t\tline += \"\\tTEXT\\n\"\n",
    "\t\twriter.write(line)\n",
    "\t\tskipped = []\n",
    "\t\tdoc_index = 0\n",
    "\t\twith open(os.path.join(train_dev_test_dir,filename), \"r\") as reader:\n",
    "\t\t\tlines = reader.readlines()\n",
    "\t\t\tfor line in lines:\n",
    "\t\t\t\tpmid, str_labels = line.strip().split(\"\\t\")\n",
    "\t\t\t\tlabels = str_labels.split(\",\")\n",
    "\t\t\t\ttext = read_text(pmid,docs_dir)\n",
    "\t\t\t\t# exclude documents w/o text\n",
    "\t\t\t\tif len(text)==0:\n",
    "\t\t\t\t\tskipped.append(doc_index)\n",
    "\t\t\t\t\tcontinue\n",
    "\t\t\t\t#print(pmid,labels,text)\n",
    "\t\t\t\tline = pmid\n",
    "\t\t\t\tfor label in all_labels:\n",
    "\t\t\t\t\tif label in labels:\n",
    "\t\t\t\t\t\tline += \"\\t1\"\n",
    "\t\t\t\t\telse:\n",
    "\t\t\t\t\t\tline += \"\\t0\"\n",
    "\t\t\t\tline += \"\\t\"+text+\"\\n\"\n",
    "\t\t\t\twriter.write(line)\n",
    "\t\t\t\tdoc_index += 1\n",
    "\twriter.close()\n",
    "\treturn tsv_file, skipped\n",
    "\n",
    "############################################\n",
    "### --------- Pre-process data --------- ###\n",
    "def pre_process_data(docs_dir,train_dev_test_dir,filename):\n",
    "\t# Import splits\n",
    "\ttsv_file, skipped = import_splits(docs_dir,train_dev_test_dir,filename)\n",
    "\t# Import data from tsv\n",
    "\tdata = pd.read_csv(tsv_file,sep='\\t')\n",
    "\t# # Select required columns\n",
    "\t# filters = []\n",
    "\t# for label in all_labels:\n",
    "\t# \tfilters.append(label)\n",
    "\t# filters.append('TEXT')\n",
    "\t# data = data[filters]\n",
    "\t# display(data)\n",
    "\t# Set your model output as categorical and save in new label col\n",
    "\tfor label in all_labels:\n",
    "\t\tif label in data:\n",
    "\t\t\tdata[label+'_label'] = pd.Categorical(data[label])\n",
    "\t# Transform your output to numeric\n",
    "\tfor label in all_labels:\n",
    "\t\tif label in data:\n",
    "\t\t\tdata[label] = data[label+'_label'].cat.codes\n",
    "\treturn data, tsv_file, skipped\n",
    "\n",
    "data_test, tsv_file_test, skipped_test = pre_process_data(papers_dir, labels_dir, \"test0.txt\")\n",
    "display(data_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "25f1a2ac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[31705587,\n",
       "        'Neurodegeneration refers to the complex process of progressive degeneration or neuronal apoptosis leading to a set of incurable and debilitating conditions. Physiologically, apoptosis is important in proper growth and development. However, aberrant and unrestricted apoptosis can lead to a variety of degenerative conditions including neurodegenerative diseases. Although dysregulated apoptosis has been implicated in various neurodegenerative disorders, the triggers and molecular mechanisms underlying such untimely and faulty apoptosis are still unknown. Hippo signaling pathway is one such apoptosis-regulating mechanism that has remained evolutionarily conserved from Drosophila to mammals. This pathway has gained a lot of attention for its tumor-suppressing task, but recent studies have emphasized the soaring role of this pathway in inflaming neurodegeneration. In addition, strategies promoting inactivation of this pathway have aided in the rescue of neurons from anomalous apoptosis. So, a thorough understanding of the relationship between the Hippo pathway and neurodegeneration may serve as a guide for the development of therapy for various degenerative diseases. The current review focuses on the mechanism of the Hippo signaling pathway, its upstream and downstream regulatory molecules, and its role in the genesis of numerous neurodegenerative diseases. The recent efforts employing the Hippo pathway components as targets for checking neurodegeneration have also been highlighted.'],\n",
       "       [30807815,\n",
       "        'The present study was aimed at broadening the profile of toxicity and biological activity of promising fused azaisocytosine-containing congeners (I-VI) possessing medical applicability and important pharmacokinetic properties. For this purpose, the in vivo zebrafish test was applied for evaluating embryotoxic effects of test compounds, whereas the ex vivo model of oxidatively-stressed rat erythrocytes was developed for assessing their antihaemolytic activities. Additionally, the MTT-based assays suitable for assessing cytotoxic and antiviral activities of I-VI were employed. The influence of compounds I-VI on zebrafish embryos/larvae was carefully investigated in relation to lack or presence of various substituents at the phenyl moiety. The least embryotoxic proved to be the parent compound (I) and its para-methyl (II) and ortho-chloro (III) derivatives. Simultaneously, they revealed the minimum embryotoxic concentrations higher than that of aciclovir, what makes them safer than this pharmaceutic. Moreover, most of test compounds showed protective effects (better or comparable to that of ascorbic acid) on oxidatively-stressed erythrocytes. All the investigated compounds were effective at inhibiting the growth of human solid tumours of pharynx (FaDu) and tongue (SCC-25). The majority of molecules showed good selectivity indices. The most selective proved to be II showing in normal Vero cells over a 5-fold and an almost 3-fold decreased cytotoxicity relative to that in tumour SCC-25 and FaDu cells, respectively. Additionally, a 3,4-dichloro derivative (VI) was shown to possess concentration-dependent inhibitory effects on the replication of Herpes simplex virus type 1 and simultaneously at active concentrations was found to be nontoxic for normal Vero cells.'],\n",
       "       [30594027,\n",
       "        \"The increase of opportunistic fungal infections raises the need for design and synthesis of new antifungal agents. Taking into account that tetrazole derivatives exhibit antifungal activity, and some of them are in the phase of clinical trials, new tetrazole derivatives bearing pyrrolidine moiety were synthesized in order to present their action mode against C.\\xa0albicans. The target compounds were obtained by N-alkylation of various 2-arylpyrrolidines with several 1-(3-chloropropyl)-5-aryl-2H-tetrazoles. Regardless of the substituents at tetrazole or pyrrolidine rings reactions took place in 48\\u202fh and with satisfactory yields ranging from 53 to 70%. We performed screen of the synthesized compounds to identify these nontoxic inhibiting the C.\\xa0albicans planktonic and sessile cells, and conducted a series of follow up studies to examine the in\\xa0vitro and in\\xa0vivo activity of the most potent antifungals. The leading antifungal inhibitor: 2-{3-[2-(3-Methylphenyl)pyrrolidin-1-yl]propyl}-5-phenyl-2H-tetrazole (3aC) and the randomly selected ones: 5-phenyl-2-[3-(2-phenylpyrrolidin-1-yl)propyl]-2H-tetrazole (3aA), 5-(4-chlorophenyl)-2-{3-[2-(4-fluorophenyl)pyrrolidin-1-yl]propyl}-2H-tetrazole (3cD), and 5-(4-chlorophenyl)-2-{3-[2-(4-chlorophenyl)pyrrolidin-1-yl]propyl}-2H-tetrazole (3cE) showed little to no toxicity against the Vero cell line and Galleria mellonella. 3aC and 3aD, the most active against biofilm in\\xa0vitro, demonstrated in\\xa0vivo activity in the invertebrate model of disseminated candidiasis. Flow cytometry analysis showed that necrotic cell death was generated under 3aC due to its interactions with the fungal membrane; this confirmed by the mitochondrial damage (XTT assay) and reduced adhesion to the TR-146\\u202fcell line at 46.05\\u202fμM. Flow cytometry was used to directly measure the redox state of the treated cells with the fluorescent DCFH probe. Pro-necrotic tetrazole derivatives (3aA, 3aC, 3cD) are unable to induce ROS production in the C.\\xa0albicans cells. Moreover, CLSM analyses revealed that the tetrazole derivatives (principally 3aC, 3aD, and 3aE) inhibit C.\\xa0albicans' ability to neutralize macrophages; a more effective phagosomes organisation was observed. 3aC's and 3aD's activity reflected in an attenuation of virulence in disseminated candidiasis in\\xa0vivo.\"],\n",
       "       [30879321,\n",
       "        'The\\xa0phospholipase C\\xa0(PLC) is a family of kinases that hydrolyze phosphatidylinositol 4,5-bisphosphate [PI(4,5)P2] to generate two second messengers, inositol 1,4,5-trisphosphate (IP3) and diacylglycerol (DAG), which stimulate distinct downstream signaling. Recently, it has been reported that PLC signaling is activated by multiple viruses for efficient replication and the virus-induced inflammatory response. In this study, we demonstrated that PLC-specific inhibitor U73122 strongly suppressed porcine reproductive and respiratory syndrome virus (PRRSV) productive infection in cell cultures. The inhibitor affected both viral post-binding cell entry and post-entry processes. The virus infection led to an early transient activation of PLCγ-1 at 0.5\\xa0h post-infection (hpi), and sustained event at a stage from 4 to 16\\xa0hpi in MARC-145 cells. In addition, U73122 inhibited the activation of p38 MAPK signaling stimulated by PRRSV infection, suggesting that PLC signaling may be associated with the virus infection-induced inflammatory response. Taken together, these studies suggested that PLC signaling played an important role in PRRSV infection or pathogenesis. Keywords: PRRSV; U73122; phospholipase C; PLCγ-1.'],\n",
       "       [30818821,\n",
       "        'Mycoplasma gallisepticum (MG) mainly infects chickens to initiate chronic respiratory disease (CRD). microRNAs (miRNAs) play vital roles according to previously reported studies. Our previous study showed that gga-miR-16-5p, in MG-infected lungs of chicken embryo, was upregulated by Illumina sequencing. The study aimed to reveal what role gga-miR-16-5p plays in CRD progression. gga-miR-16-5p was upregulated in MG-infected fibroblast cells (DF-1). Phosphoinositide-3-kinase regulatory subunit 1 (PIK3R1) was demonstrated as the target gene of gga-miR-16-5p. Furthermore, PIK3R1 expression was lower in MG-infected groups than it in noninfected controls measured by qPCR. Additionally, overexpressed gga-miR-16-5p could downregulate PIK3R1 and phosphorylated serine/threonine kinase (p-Akt) to express protein, whereas there is an opposite effect on inhibition. Overexpressed gga-miR-16-5p resulted in decreased activity of tumor necrosis factor alpha (TNF-α) and the nuclear factor-kappaB (NF-κB) by qPCR. Furthermore, overexpressed gga-miR-16-5p restricted cell multiplication, cycle progression, and increased apoptosis of MG-infected DF-1 cells, whereas inhibited gga-miR-16-5p led to the opposite effect. Collectively, upregulated gga-miR-16-5p could decrease multiplication, cycle progression, and increase apoptosis of MG-infected DF-1 cells, at least partly through directly targeting PIK3R1 and inhibiting PI3K/Akt/NF-κB pathway to exert an anti-inflammatory effect. Our results will provide more experimental evidence to bring pathogenesis of MG infection to light.'],\n",
       "       [30537062,\n",
       "        'The production of piscine viruses, in particular of koi herpesvirus (KHV, CyHV-3) and infectious salmon anaemia virus (ISAV), is still challenging due to the limited susceptibility of available cell lines to these viruses. A number of cell lines from different fish species were compared to standard diagnostic cell lines for KHV and ISAV regarding their capability to exhibit a cytopathic effect (CPE) and to accumulate virus. Two cell lines, so far undescribed, appeared to be useful for diagnostic purposes. Fr994, a cell line derived from ovaries of rainbow trout (Oncorhynchus mykiss), produced constantly high ISA virus (ISAV) titres and developed a pronounced CPE even at high cell passage numbers, while standard cell lines are reported to gradually loose these properties upon propagation. Another cell line isolated from the head kidney of common carp (Cyprinus carpio), KoK, showed a KHV induced CPE earlier than the standard cell line used for diagnostics. A third cell line, named Fin-4, established from the fin epithelium of rainbow trout did not promote efficient replication of tested viruses, but showed antigen sampling properties and might be useful as an in vitro model for virus uptake or phagocytosis.'],\n",
       "       [30955825,\n",
       "        'Porcine reproductive and respiratory syndrome (PRRS) is a severe respiratory disease that leads to huge economic losses in the pig industry throughout the world. Although there are several vaccines available, the protective efficacy is limited. Therefore, new control strategies to prevent PRRS virus (PRRSV) infection are urgently required. We have previously reported that CH25H and 25HC can significantly inhibit the replication of PRRSV by preventing viral entry. In the present study, we found that 25HC with a low IC50 value significantly decreased the replication of different PRRSV strains, and increased the production of IL-1β and IL-8 in porcine primary alveolar macrophages and the lung tissue. In pigs challenged with highly pathogenic PRRSV, treatment with 25HC was associated with an obvious reduction in the level of viremia and viral load in lung samples and nasal swabs, as well as decreased lung injury and an increased survival rate. These findings suggest that 25HC could be a promising antiviral drug against PRRSV in the future.'],\n",
       "       [27391329,\n",
       "        'The present research was designed to investigate whether endothelin-1 (ET-1) secretion can be induced by oxyhemoglobin and whether nuclear factor κB (NF-κB) is involved in the regulation of ET-1 transcription in cerebrovascular muscle cells. Cerebrovascular muscle cells isolated from a rabbit basilar artery were stimulated by oxyhemoglobin (OxyHb) and ET-1 production was increased significantly in the supernatant. Inhibition of NF-κB with pyrrolidine dithiocarbamate and small interfering RNA decreased the expression of ET-1. Nuclear translocation of NF-κB and the degradation of IkB-α was observed with the stimulation of OxyHb. The supernatant obtained from cerebrovascular muscle cells stimulated by OxyHb produced contractions in arterial rings and was blocked by the ET-1 receptor antagonist (BQ-123). The time course of the OxyHb-induced contractions of the basilar artery rings correlated with the time course of the OxyHb-induced ET-1 secretion. The contraction of the basilar artery rings induced by OxyHb was attenuated when the artery rings were preincubated with pyrrolidine dithiocarbamate and SN50 (20 and 10\\u2009µM, respectively). These results indicate that cerebrovascular muscle cells may be an important source of ET-1 production after subarachnoid hemorrhage. NF-κB was involved in the expression of ET-1 and the inhibition of the NF-κB pathway may be beneficial for the treatment of cerebral vasospasm.'],\n",
       "       [27651133,\n",
       "        'Data obtained in both animal models and clinical trials suggest that cell-based therapies represent a potential therapeutic strategy for lung repair and remodeling. Recently, new therapeutic approaches based on the use of stem cell derivatives (e.g., conditioned medium (CM) and microvesicles (MVs)) to regenerate tissues and improve their functions were proposed. The aim of this study was to investigate the immunomodulatory effects of equine amniotic mesenchymal cell derivatives on lipopolysaccharide (LPS)-induced cytokine production in equine alveolar macrophages, which may be beneficial in lung inflammatory disorders such as recurrent airway obstruction (RAO) in horses. RAO shares many features with human asthma, including an increased number of cells expressing mRNA for interleukin (IL)-4 and IL-5 and a decreased expression of IFN-γ in bronchoalveolar lavage fluid (BALF) of affected horses. The release of TNF-α, IL-6, and TGF-β1 at different time points (1, 24, 48, and 72\\xa0h) was measured in equine alveolar macrophages stimulated or not with LPS (10 and 100\\xa0ng/mL) in the presence or absence of 10\\xa0% CM or 50\\u2009×\\u200910(6) MVs/mL. Cytokines were measured using commercially available ELISA kits. For multiple comparisons, analysis of variance was used with Tukey post-hoc test. Differences were considered significant at p\\u2009≤\\u20090.05. Significant modulatory effects of CM on LPS-induced TNF-α release at 24\\xa0h, and of both CM and MVs on TNF-α release at 48\\xa0h were observed. A trend toward a modulatory effect of both CM and MVs on the release of TGF-β and possibly IL-6 was visible over time. Results support the potential use of CM and MVs in lung regenerative medicine, especially in situations in which TGF-β may be detrimental, such as respiratory allergy. Further studies should evaluate the potential clinical applications of CM and MVs in equine lung diseases, such as RAO and other inflammatory disorders.'],\n",
       "       [28650243,\n",
       "        'OBJECTIVE To determine whether major histocompatability complex (MHC) class II expression in equine mesenchymal stem cells (MSCs) changes with exposure to a proinflammatory environment reflective of an inflamed joint. SAMPLE Cryopreserved bone marrow-derived MSCs from 12 horses and cartilage and synovium samples from 1 horse euthanized for reasons other than lameness. PROCEDURES In part 1 of a 3-part study, the suitability of a quantitative reverse transcriptase PCR (qRT-PCR) assay for measurement of MHC class II expression in MSCs following stimulation with interferon (IFN)-γ was assessed. In part 2, synoviocyte-cartilage cocultures were or were not stimulated with interleukin (IL)-1β (10 ng/mL) to generate conditioned media that did and did not (control) mimic an inflamed joint environment. In part 3, a qRT-PCR assay was used to measure MSC MHC class II expression after 96 hours of incubation with 1 of 6 treatments (control-conditioned medium, IL-1β-conditioned medium, and MSC medium alone [untreated control] or with IL-1β [10 ng/mL], tumor necrosis factor-α [10 ng/mL], or IFN-γ [100 ng/mL]). RESULTS The qRT-PCR assay accurately measured MHC class II expression. Compared with MHC class II expression for MSCs exposed to the untreated control medium, that for MSCs exposed to IL-1β was decreased, whereas that for MSCs exposed to IFN-γ was increased. Neither the control-conditioned nor tumor necrosis factor-α medium altered MHC class II expression. CONCLUSIONS AND CLINICAL RELEVANCE Results indicated that MSC exposure to proinflammatory cytokine IL-1β decreased MHC class II expression and antigenicity. Treatment of inflamed joints with allogeneic MSCs might not be contraindicated, but further investigation is warranted.'],\n",
       "       [29617946,\n",
       "        'Muscle satellite cells (SC) are resident stem-like cells that play an integral role in skeletal muscle growth and repair. Understanding how SC maintain their identities and dynamic properties is critical to animal growth. However, the genetic and environmental factors governing SC behaviors and the underpinning mechanisms remain unknown. To explore whether genetic selection influences SC behaviors, we used 2 lines of chickens selected for over 50 generations with over a 10-fold difference in body weight at 56 d of age-the Virginia high weight selection (HWS) and low weight selection (LWS) lines. To study these 2 lines, we performed both in vivo and in vitro experiments. In vivo, we studied the abundance of SC in normal physiological settings and tested their functional roles in muscle regeneration using a muscle injury model. In vitro, we isolated SC from chicken skeletal muscle and assayed their ability to proliferate and differentiate under cultured conditions. Immunohistochemical staining of breast muscle (pectoralis major) revealed that muscle fibers from HWS chickens possessed more SC than those from LWS. Further analysis showed that the SC pool from HWS muscles contained a higher percentage of activated SC compared to that of LWS. When isolated SC from HWS and LWS muscles were cultured, HWS SC exhibited greater abilities to proliferate and differentiate than those SC from LWS. To test whether the observed in vitro differences in SC properties could be confirmed in vivo, we subjected chicken breast muscle to barium chloride to induce muscle injury and regeneration. Consistent with in vitro data, breast muscle in HWS chicken experienced a faster and more robust recovery than that of LWS, as evidenced by quicker regeneration and larger muscle fiber size. Taken together, these findings suggest divergent selection for body weight not only results in correlated responses in SC number, but also changes SC growth kinetics. Further dissection of the molecular mechanism will aid the identification of the target molecules for growth intervention in chickens.'],\n",
       "       [31167637,\n",
       "        \"Intra-tumor heterogeneity is known to contribute to cancer complexity and drug resistance. Understanding the number of distinct subclones and the evolutionary relationships between them is scientifically and clinically very important and still a challenging problem. In this paper, we present BAMSE (BAyesian Model Selection for tumor Evolution), a new probabilistic method for inferring subclonal history and lineage tree reconstruction of heterogeneous tumor samples. BAMSE uses somatic mutation read counts as input and can leverage multiple tumor samples accurately and efficiently. In the first step, possible clusterings of mutations into subclones are scored and a user defined number are selected for further analysis. In the next step, for each of these candidates, a list of trees describing the evolutionary relationships between the subclones is generated. These trees are sorted by their posterior probability. The posterior probability is calculated using a Bayesian model that integrates prior belief about the number of subclones, the composition of the tumor and the process of subclonal evolution. BAMSE also takes the sequencing error into account. We benchmarked BAMSE against state of the art software using simulated datasets. In this work we developed a flexible and fast software to reconstruct the history of a tumor's subclonal evolution using somatic mutation read counts across multiple samples. BAMSE software is implemented in Python and is available open source under GNU GLPv3 at https://github.com/HoseinT/BAMSE .\"],\n",
       "       [30427935,\n",
       "        'Sexually reproducing parasites, such as malaria parasites, experience a trade-off between the allocation of resources to asexual replication and the production of sexual forms. Allocation by malaria parasites to sexual forms (the conversion rate) is variable but the evolutionary drivers of this plasticity are poorly understood. We use evolutionary theory for life histories to combine a mathematical model and experiments to reveal that parasites adjust conversion rate according to the dynamics of asexual densities in the blood of the host. Our model predicts the direction of change in conversion rates that returns the greatest fitness after perturbation of asexual densities by different doses of antimalarial drugs. The loss of a high proportion of asexuals is predicted to elicit increased conversion (terminal investment), while smaller losses are managed by reducing conversion (reproductive restraint) to facilitate within-host survival and future transmission. This non-linear pattern of allocation is consistent with adaptive reproductive strategies observed in multicellular organisms. We then empirically estimate conversion rates of the rodent malaria parasite Plasmodium chabaudi in response to the killing of asexual stages by different doses of antimalarial drugs and forecast the short-term fitness consequences of these responses. Our data reveal the predicted non-linear pattern, and this is further supported by analyses of previous experiments that perturb asexual stage densities using drugs or within-host competition, across multiple parasite genotypes. Whilst conversion rates, across all datasets, are most strongly influenced by changes in asexual density, parasites also modulate conversion according to the availability of red blood cell resources. In summary, increasing conversion maximises short-term transmission and reducing conversion facilitates in-host survival and thus, future transmission. Understanding patterns of parasite allocation to reproduction matters because within-host replication is responsible for disease symptoms and between-host transmission determines disease spread.'],\n",
       "       [28473453,\n",
       "        'The hard tick Ixodes ricinus is an important disease vector whose salivary secretions mediate blood-feeding success on vertebrate hosts, including humans. Here we describe the expression profiles and downstream analysis of de novo-discovered microRNAs (miRNAs) expressed in I. ricinus salivary glands and saliva. Eleven tick-derived libraries were sequenced to produce 67,375,557 Illumina reads. De novo prediction yielded 67 bona fide miRNAs out of which 35 are currently not present in miRBase. We report for the first time the presence of microRNAs in tick saliva, obtaining furthermore molecular indicators that those might be of exosomal origin. Ten out of these microRNAs are at least 100 times more represented in saliva. For the four most expressed microRNAs from this subset, we analyzed their combinatorial effects upon their host transcriptome using a novel in silico target network approach. We show that only the inclusion of combinatorial effects reveals the functions in important pathways related to inflammation and pain sensing. A control set of highly abundant microRNAs in both saliva and salivary glands indicates no significant pathways and a far lower number of shared target genes. Therefore, the analysis of miRNAs from pure tick saliva strongly supports the hypothesis that tick saliva miRNAs can modulate vertebrate host homeostasis and represents the first direct evidence of tick miRNA-mediated regulation of vertebrate host gene expression at the tick-host interface. As such, the herein described miRNAs may support future drug discovery and development projects that will also experimentally question their predicted molecular targets in the vertebrate host.'],\n",
       "       [28830547,\n",
       "        'The patterns of relative species abundance are commonly studied in ecology and epidemiology to provide insights into underlying dynamical processes. Molecular types (MVLA-types) of Mycobacterium bovis, the causal agent of bovine tuberculosis, are now routinely recorded in culture-confirmed bovine tuberculosis cases in Northern Ireland. In this study, we use ecological approaches and simulation modelling to investigate the distribution of relative abundances of MVLA-types and its potential drivers. We explore four biologically plausible hypotheses regarding the processes driving molecular type relative abundances: sampling and speciation; structuring of the pathogen population; historical changes in population size; and transmission heterogeneity (superspreading). Northern Irish herd-level MVLA-type surveillance shows a right-skewed distribution of MVLA-types, with a small number of types present at very high frequencies and the majority of types very rare. We demonstrate that this skew is too extreme to be accounted for by simple neutral ecological processes. Simulation results indicate that the process of MVLA-type speciation and the manner in which the MVLA-typing loci were chosen in Northern Ireland cannot account for the observed skew. Similarly, we find that pathogen population structure, assuming for example a reservoir of infection in a separate host, would drive the relative abundance distribution in the opposite direction to that observed, generating more even abundances of molecular types. However, we find that historical increases in bovine tuberculosis prevalence and/or transmission heterogeneity (superspreading) are both capable of generating the skewed MVLA-type distribution, consistent with findings of previous work examining the distribution of molecular types in human tuberculosis. Although the distribution of MVLA-type abundances does not fit classical neutral predictions, our simulations show that increases in pathogen population size and/or superspreading are consistent with the pattern observed, even in the absence of selective pressures acting on the system.'],\n",
       "       [31174159,\n",
       "        'To understand the structural and functional importance of PIK3CA somatic mutations, whole exome sequencing, molecular dynamics simulation techniques in combination with in silico prediction algorithms such as SIFT, PolyPhen, Provean and CADD were employed. Twenty out of eighty missense somatic mutations in PIK3CA gene were found to be pathogenic by all the four algorithms. Most recurrent mutations found were known hotspot PIK3CA mutations with known clinical significance like p.E545\\u202fK, p.E545A, p.E545\\u202fG and p.C420R. A missense mutation p.G118D was found to be recurrently mutated in 5 cases. Interestingly, this mutation was observed in one of the patients who underwent whole exome sequencing and was completely absent from the controls. To see the effect of this mutation on the structure of PIK3CA protein, molecular dynamics simulation was performed. By molecular dynamics approach, we have shown that p.G118D mutation deviated from the native structure which was supported by the decrease in the number of hydrogen bonds, difference in hydrogen bond distance and angle, difference in root mean square deviation between the native and the mutant structures.'],\n",
       "       [29605007,\n",
       "        \"Although various studies have provided novel insights into how to best design, analyze and interpret a fecal egg count reduction test (FECRT), it is still not straightforward to provide guidance that allows improving both the standardization and the analytical performance of the FECRT across a variety of both animal and nematode species. For example, it has been suggested to recommend a minimum number of eggs to be counted under the microscope (not eggs per gram of feces), but we lack the evidence to recommend any number of eggs that would allow a reliable assessment of drug efficacy. Other aspects that need further research are the methodology of calculating uncertainty intervals (UIs; confidence intervals in case of frequentist methods and credible intervals in case of Bayesian methods) and the criteria of classifying drug efficacy into 'normal', 'suspected' and 'reduced'. The aim of this study is to provide complementary insights into the current knowledge, and to ultimately provide guidance in the development of new standardized guidelines for the FECRT. First, data were generated using a simulation in which the 'true' drug efficacy (TDE) was evaluated by the FECRT under varying scenarios of sample size, analytic sensitivity of the diagnostic technique, and level of both intensity and aggregation of egg excretion. Second, the obtained data were analyzed with the aim (i) to verify which classification criteria allow for reliable detection of reduced drug efficacy, (ii) to identify the UI methodology that yields the most reliable assessment of drug efficacy (coverage of TDE) and detection of reduced drug efficacy, and (iii) to determine the required sample size and number of eggs counted under the microscope that optimizes the detection of reduced efficacy. Our results confirm that the currently recommended criteria for classifying drug efficacy are the most appropriate. Additionally, the UI methodologies we tested varied in coverage and ability to detect reduced drug efficacy, thus a combination of UI methodologies is recommended to assess the uncertainty across all scenarios of drug efficacy estimates. Finally, based on our model estimates we were able to determine the required number of eggs to count for each sample size, enabling investigators to optimize the probability of correctly classifying a theoretical TDE while minimizing both financial and technical resources.\"],\n",
       "       [28091882,\n",
       "        'The surface-associated proteins play a key role in bacterial physiology and pathogenesis, and are the major targets in the development of new vaccines. These proteins contribute to the adaptation of bacteria to different hosts and environments. To study differences at the genomic level, we first sequenced the whole genome of Streptococcus iniae from fish (IUSA-1 strain) and compared it to Streptococcus iniae from human (9117 strain), revealing a high similitude between both strains. To gain further insights into host- and environment-specific differences, we then studied proteins in silico and by High Performance Liquid Chromatography. This approach successfully identified 54 secreted and surface proteins, including several proteins involved in cell wall synthesis and transport of solutes, as well as proteins with yet unknown function. These proteins highlight as interesting targets for further investigation in the interaction between Streptococcus iniae and its environment. Results reported in this study have shown a first analysis about the predicted and experimental associated proteins of Streptococcus iniae isolated from two different hosts: human and fish.'],\n",
       "       [22289896,\n",
       "        'To examine the relative susceptibility of cartilage and meniscus tissues to mechanical injury by applying a single, controlled overload and observing cellular, biochemical, and mechanical changes. Cartilage and meniscus tissue explants in radial confinement were subjected to a range of injury by indenting to 40% strain at three different strain rates: 0.5%/s (slow), 5%/s (medium), or 50%/s (fast). Following injury, samples were cultured for either 1 or 9 days. Explants were assayed for cell metabolic activity, water content, and sulfated glycosaminoglycan (sGAG) content. Mechanical properties of explants were determined in torsional shear and unconfined compression. Conditioned medium was assayed for sGAG and lactate dehydrogenase (LDH) release. Peak injury force increased with strain rate but both tissues displayed little to no macroscopic damage. Cell metabolism was lowest in medium and fast groups on day 1. Cell lysis increased with peak injury force and loading rate in both tissues. In contrast, sGAG content and release did not significantly vary with loading rate. Additionally, mechanical properties did not significantly vary with loading rate in either tissue. By use of a custom confinement chamber, large peak forces were obtained without macroscopic destruction of the explants. At the loads achieved in this studied, cell damage was induced without detectable physical or compositional changes. These results indicate that sub-failure injury can induce biologic damage that may not be readily detected and could be an early event in osteoarthritis genesis.'],\n",
       "       [24681021,\n",
       "        'Sinoatrial node (SAN) dysfunction increases the occurrences of atrial fibrillation (AF). The pulmonary veins (PVs) play a critical role in the pathophysiology of AF. The purpose of this study was to evaluate whether SAN electrical activity can modulate PV arrhythmogenesis. Conventional microelectrodes and multi-electrode array system were used to simultaneously record the electrical activity and conduction properties of rabbit SAN and PV tissue preparations with and without SAN-PV interruptions before and after perfusion with Anemonia sulcata toxin (ATX)-II (100 nM) or isoproterenol (1 μM). ATX-II significantly increased PV beating rates, which overdrove SAN electrical activity with the occurrences of PV burst firings in 5 (56%) of 9 tissue preparations, and induced SAN-PV conduction block in 6 (67%) of 9 preparations. After SAN-PV disconnection, ATX-II induced burst firing and early afterdepolarizations in 8 (89%) of 9 PVs. Moreover, the multi-electrode array found that ATX-II reversed the electrical conduction between the SAN and PV with an increase in electrical activity from 1.8 ± 0.6 to 2.9 ± 0.6 Hz (P<0.05) in SAN-PV preparations (n=7). In contrast, isoproterenol did not reverse electrical conduction between the SAN and PV with an increase in electrical activity from 1.8 ± 0.2 to 3.0 ± 0.3 Hz (P<0.005) in SAN-PV preparations (n=7). SAN electrical activity modulates PV arrhythmogenesis. SAN-PV conduction blocks can increase PV arrhythmogenesis.'],\n",
       "       [27915442,\n",
       "        'The global incidence of Fusarium head blight and attendant cereal grains multi-contamination by the trichothecene mycotoxins deoxynivalenol (DON) and nivalenol (NIV) are increasing as a possible result of climate change and inadequate agricultural practices. At the molecular level, these mycotoxins bind to the ribosome, activate the mitogen-activated protein kinase and induce a local and systemic inflammation. DON is of public health concern owing to the narrow margin between exposure and tolerable daily intake. The intestinal inflammatory response to DON, NIV and their mixture was analyzed to determine thresholds for their intestinal pro-inflammatory effects and characterize the type and magnitude of their interaction. Fully differentiated three-dimensional porcine jejunal explants were exposed to increasing doses of DON and NIV alone or in combination; the expression levels of IL-1α, IL-1β, IL-8, IL-17A and IL-22 were measured by RT-PCR. Doses as low as 0.16\\xa0µM DON or 0.73\\xa0µM NIV significantly increase the intestinal expression levels of the tested inflammation-related genes. These doses are lower than those previously reported for other intestinal toxicity endpoints. The combined pro-inflammatory activity of DON and NIV was synergistic for all the tested genes with combination index value range of 0.23-0.8. Our results indicate that (1) inflammation is a very sensitive endpoint for the intestinal toxicity of the trichothecenes and (2) co-exposure to DON and NIV has a greater inflammatory effect than induced by mycotoxins alone. This synergy should be taken into account considering the frequent co-occurrence of DON and NIV in the diet.'],\n",
       "       [30085372,\n",
       "        'Decreased fertility associated with maternal ageing is a well-known critical problem, and progesterone (P4) concentration decreases during the menopause transition in women. The corpus luteum (CL) secretes P4, thereby supporting the implantation and maintenance of pregnancy. It is proposed that a bovine model is suitable for studying age-associated decline of fertility in women because the physiology of cows is similar to that of women and cows have a greater longevity compared with other animal models. Thus, we investigated the age-dependent qualitative changes and inflammatory responses in the bovine CL. In vivo experiment: Cows were divided into three groups, namely, young (mean age: 34.8\\xa0months), middle (80.1\\xa0months) and aged (188.9\\xa0months). Blood samples were collected on days 7 and 12 during the estrous cycle. In vitro experiments: Cows were divided into young (mean age: 27.6\\xa0months) and aged (183.1\\xa0months). The CL tissues of these groups were collected from a local slaughterhouse and used for tissue culture experiments. An in vivo experiment, plasma P4 concentration in aged cows was significantly lower than that in young cows, whereas no difference was found regarding the area of CL. An in vitro examination in the bovine CL tissues showed that the luteal P4 concentration, P4 secretion, and mRNA expression of StAR and 3β-HSD were lower in aged cows compared with young cows, especially in the early luteal phase. However, no differences were detected in the mRNA expression of inflammation- and senescence-related factors and inflammatory responses to lipopolysaccharides between the CL tissues from young and aged cows, indicating that an age-dependent increase in inflammation is not involved in the luteal function. P4 production and secretion from the bovine CL diminish in old cows, especially during the early luteal phase, suggesting that senescence may affect the luteal function in cows.'],\n",
       "       [23365272,\n",
       "        'Surgical removal of porcine intestinal tissue followed by an ex vivo challenge is an alternative technique of testing the anti-inflammatory effect of bioactive compounds in the intestine of live pigs. We investigated the effects of ex vivo incubation of porcine colonic tissue on the quantity and quality of total RNA over a 12-h time period. Colonic tissue of pig (n = 6) was surgically removed immediately postslaughter and incubated for 0, 3, 6, and 12 h in a humidified cell culture incubator with 5% CO(2) at 37°C. Tissue samples were processed for RNA extraction. The quantity and quality of total RNA were assessed on a NanoDrop Spectrophotometer and an Agilent 2100 Bioanalyzer, respectively. Ex vivo incubation had an effect on both the quantity (P < 0.001) and quality (P < 0.001) of total RNA. Relative to the RNA yield at 0 h (505.0 ± 48.64 μg/mg), the yield was significantly reduced after 6 h (227.6 ± 25.52 μg/mg; P < 0.001) and 12 h (159.3 ± 24.19 μg/mg; P < 0.001) of incubation. The 28S and 18S rRNA bands were visibly intact after 0, 3, and 6 h of incubation. However, after 12 h of incubation, a degraded RNA profile was evident. The RNA integrity number (RIN) values for the 0, 3, 6, and 12 h of incubation were 9.4 ± 0.10, 9.0 ± 0.10, 6.7 ± 0.17 (P < 0.001), and 3.3 ± 0.24 (P < 0.001), respectively. The transcript abundances of 4 constitutively expressed genes glyceraldehyde 3-phosphate dehydrogenase (GAPDH), beta actin (ACTB), beta 2-microglobulin (B2M), and peptidylprolyl isomerase A (PPIA) were reduced at both 6 and 12 h of incubation. It is concluded that ex vivo incubation of porcine colonic tissue up to 3 h postmortem generates good quality total RNA suitable for gene expression studies.'],\n",
       "       [29070686,\n",
       "        'Infectious laryngotracheitis virus (ILTV) is an alphaherpesvirus that infects chickens, causing upper respiratory tract disease and significant losses to poultry industries worldwide. Glycoprotein G (gG) is a broad-range viral chemokine-binding protein conserved among most alphaherpesviruses, including ILTV. A number of studies comparing the immunological parameters between infection with gG-expressing and gG-deficient ILTV strains have demonstrated that expression of gG is associated with increased virulence, modification of the amount and the composition of the inflammatory response, and modulation of the immune responses toward antibody production and away from cell-mediated immune responses. The aims of the current study were to examine the establishment of infection and inflammation by ILTV and determine how gG influences that response to infection. In vitro infection studies using tracheal organ tissue specimen cultures and blood-derived monocytes and in vivo infection studies in specific-pathogen-free chickens showed that leukocyte recruitment to the site of infection is an important component of the induced pathology and that this is influenced by the expression of ILTV gG and changes in the transcription of the chicken orthologues of mammalian CXC chemokine ligand 8 (CXCL8), chicken CXCLi1 and chicken CXCLi2, among other cytokines and chemokines. The results from this study demonstrate that ILTV gG interferes with chemokine and cytokine transcription at different steps of the inflammatory cascade, thus altering inflammation, virulence, and the balance of the immune response to infection.IMPORTANCE Infectious laryngotracheitis virus is an alphaherpesvirus that expresses gG, a conserved broad-range viral chemokine-binding protein known to interfere with host immune responses. However, little is known about how gG modifies virulence and influences the inflammatory signaling cascade associated with infection. Here, data from in vitro and in vivo infection studies are presented. These data show that gG has a direct impact on the transcription of cytokines and chemokine ligands in vitro (such as chicken CXCL8 orthologues, among others), which explains the altered balance of the inflammatory response that is associated with gG during ILTV infection of the upper respiratory tract of chickens. This is the first report to associate gG with the dysregulation of cytokine transcription at different stages of the inflammatory cascade triggered by ILTV infection of the natural host.'],\n",
       "       [24001121,\n",
       "        'Swine have been regarded as intermediate hosts in the spread of influenza from birds to humans but studies of the sialylated glycans that comprise their respiratory tract have not been extensively studied in the past. This study analyzed the sialylated N-glycan and O-glycan profile of swine trachea and lung and correlated this with ex-vivo infection of swine explants with avian influenza viruses. Lungs and tracheal samples were obtained from normal farm and laboratory raised swine and used for ex vivo infection as well as mass spectrometric analysis. Infection of the ex vivo tissues used high pathogenic and low pathogenic avian viruses including the novel H7N9 virus that emerged in China in early 2013. Assessment of successful replication was determined by TCID50 as well as virus immunohistochemistry. The N-glycan and O-glycan profiles were measured by MALDI-TOF and sialylated linkages were determined by sialidase treatment. Lectin binding histochemistry was also performed on formalin fixed tissue samples with positive binding detected by chromogen staining. The swine respiratory tract glycans differed from the human respiratory tact glycans in two main areas. There was a greater abundance of Gal-α-Gal linkages resulting in a relative decrease in sialylated glycans. The swine respiratory tract also had a greater proportion of glycans containing Neu5Gc and Siaα2-6 glycans than the human respiratory tract. Infection with avian viruses was confined primarily to lung bronchioles rather than trachea and parenchyma. In contrast to previous studies we found that there was not as much expression of Siaα2-3 glycans on the surface of the trachea. Infection of Siaα2-3 binding avian viruses was restricted to the lower respiratory tract bronchioles. This finding may diminish the ability of the swine to act as an intermediary in the transmission of avian viruses to humans.'],\n",
       "       [29992618,\n",
       "        'Bone marrow-derived mesenchymal stem cells (BM-MSCs) are frequently used in the treatment of musculoskeletal injuries. Fully characterised cells that are readily available for use is optimum. Allogenic BM-MSCs can satisfy the need for rapid treatment, however, their safety has been questioned. Objectives were to characterise BM-MSCs from an adult donor horse, in vitro, and to identify and describe adverse reactions that occurred following their injection into other horses. We hypothesised that BM-MSCs capable of proliferation, differentiation and lacking MHC II from one donor could be implanted into another individual without significant adverse reactions and the frequency of adverse reactions in clinical cases would be similar to that previously reported for autologous BM-MSCs. Retrospective clinical study. BM-MSCs were proliferated and characterised from one donor and cryopreserved for clinical use. Medical records for horses injected with allogenic BM-MSCs from this donor at a single hospital were used. After routine lameness exam, lesions were identified using diagnostic ultrasound or MRI. Post injection reaction was defined as increased pain, swelling, or heat at or near injection site, or increased lameness. Treatments required for each reaction were noted. BM-MSCs proliferated and underwent differentiation. Cells were found to be negative for MHC-II (<2%) and were viable after cryopreservation and shipping. Ten of 230 (4.35%) injections were noted to be associated with an adverse reaction. Adverse reactions occurred in synovial structures (n = 3) and in soft tissues (n = 7). This investigation could underestimate the number and severity of reactions. Mild reactions, such as synovitis, may have been missed. Also, anti-inflammatory drugs could overshadow mild reactions, making them less likely to be detected. Fully characterised allogenic BM-MSCs originating from a single donor horse can be administered to horses with soft tissue injuries with a low rate of adverse reaction. The Summary is available in Portuguese - see Supporting Information.'],\n",
       "       [31369405,\n",
       "        'The aim of this study was to provide a retrospective analysis of the presentation, demographics, and treatment regimens for ocular toxoplasmosis at a large tertiary referral uveitis center. Retrospective cohort study. A total of 48 patients with ocular toxoplasmosis who presented to Sydney Eye Hospital participated in this study. This is a retrospective review of patient files who presented to Sydney Eye Hospital between 2007 and 2016 with clinical features consistent with ocular toxoplasmosis. Baseline risk factors and treatment details were recorded and analyzed. Main outcome measures were visual acuity and relapse rate compared with other studies in ocular toxoplasmosis. The median age was 35.5 (interquartile range 21-50) with 30 (60%) patients having no previous symptomatic episodes or evidence of chorioretinal scarring. Visual acuity at presentation was 0.51 or 6/19 (SE 0.096) and at follow-up 0.31 or 6/12 (SE 0.094). Nine patients experienced a recurrence during the period of observation with median time to recurrence 2.2 years (SE 0.45) and the relapse rate was 0.09/person-years. Location of lesion was predominantly within the vascular arcades (n\\u200a=\\u200a44) with macular involvement in 9 patients. Most patients received clindamycin therapy (n\\u200a=\\u200a34) with pyrimethamine and sulfadiazine was used for those with macula involvement. Patients with ocular toxoplasmosis had fewer recurrences compared with other published series and had better visual recovery. The majority of patients received clindamycin and oral prednisolone which were well tolerated with pyrimethazine and sulfadiazine reserved for those with macula-involving disease.'],\n",
       "       [31379463,\n",
       "        'Phlebotomy is an error-prone process in which mistakes are difficult to reveal. This case report describes the effect on laboratory results originating from a blood sample collected in close proximity to an intravenous catheter. A 69-year-old male patient was referred to the Emergency department where pneumonia was suspected. Phlebotomy was performed to collect blood samples to assess electrolytes, renal function, liver function, infection and haematological parameters. The laboratory analysis showed reduced potassium and calcium concentrations. To prevent life-threatening cardiac failure the clinician decided to correct those electrolytes. Remarkably, the electrocardiogram showed no abnormalities corresponding to hypokalaemia and hypocalcaemia. This observation, in combination with an overall increase in laboratory parameters with the exception of sodium and chloride, led to the suspicion of a preanalytical error. Retrospectively, an intravenous catheter was inserted in close proximity of the puncture place but no continuous infusion was started prior to phlebotomy. However, the intravenous catheter was flushed with sodium chloride. Since potential other causes were excluded, the flushing of the intravenous catheter with sodium chloride prior to phlebotomy was the most probable cause for the deviating laboratory results and subsequently for the unnecessary potassium and calcium suppletion. This case underlines the importance of caution in the interpretation of laboratory results obtained from specimens that are collected in the proximity of an intravenous catheter, even in the absence of continuous infusion.'],\n",
       "       [28926595,\n",
       "        'The aim of this study was to define the microbiota of water buffalo milk during sub-clinical and clinical mastitis, as compared to healthy status, by using high-throughput sequencing of the 16S rRNA gene. A total of 137 quarter samples were included in the experimental design: 27 samples derived from healthy, culture negative quarters, with a Somatic Cell Count (SCC) of less than 200,000 cells/ml; 27 samples from quarters with clinical mastitis; 83 samples were collected from quarters with subclinical mastitis, with a SCC number greater of 200,000 cells/ml and/or culture positive for udder pathogens, without clinical signs of mastitis. Bacterial DNA was purified and the 16S rRNA genes were individually amplified and sequenced. Significant differences were found in milk samples from healthy quarters and those with sub-clinical and clinical mastitis. The microbiota diversity of milk from healthy quarters was richer as compared to samples with sub-clinical mastitis, whose microbiota diversity was in turn richer as compared to those from clinical mastitis. The core microbiota of water buffalo milk, defined as the asset of microorganisms shared by all healthy milk samples, includes 15 genera, namely Micrococcus, Propionibacterium, 5-7N15, Solibacillus, Staphylococcus, Aerococcus, Facklamia, Trichococcus, Turicibacter, 02d06, SMB53, Clostridium, Acinetobacter, Psychrobacter and Pseudomonas. Only two genera (Acinetobacter and Pseudomonas) were present in all the samples from sub-clinical mastitis, and no genus was shared across all in clinical mastitis milk samples. The presence of mastitis was found to be related to the change in the relative abundance of genera, such as Psychrobacter, whose relative abundance decreased from 16.26% in the milk samples from healthy quarters to 3.2% in clinical mastitis. Other genera, such as SMB53 and Solibacillus, were decreased as well. Discriminant analysis presents the evidence that the microbial community of healthy and clinical mastitis could be discriminated on the background of their microbiota profiles.'],\n",
       "       [31682953,\n",
       "        'Progressive myoclonic epilepsies (PMEs) comprise a group of rare disorders of different genetic aetiologies, leading to childhood-onset myoclonus, myoclonic seizures and subsequent neurological decline. One of the genetic causes for PME, a mutation in the gene coding for Golgi SNAP receptor 2 (GOSR2), gives rise to a PME-subtype prevalent in Northern Europe and hence referred to as North Sea Progressive Myoclonic Epilepsy (NS-PME). Treatment for NS-PME, as for all PME subtypes, is symptomatic; the pathophysiology of NS-PME is currently unknown, precluding targeted therapy. Here, we investigated the pathophysiology of NS-PME. By means of chart review in combination with interviews with patients (n\\u202f=\\u202f14), we found heat to be an exacerbating factor for a majority of NS-PME patients (86%). To substantiate these findings, we designed a NS-PME Drosophila melanogaster model. Downregulation of the Drosophila GOSR2-orthologue Membrin leads to heat-induced seizure-like behaviour. Specific downregulation of GOSR2/Membrin in glia but not in neuronal cells resulted in a similar phenotype, which was progressive as the flies aged and was partially responsive to treatment with sodium barbital. Our data suggest a role for GOSR2 in glia in the pathophysiology of NS-PME.'],\n",
       "       [31374061,\n",
       "        'We report a rare case of neurofibroma in the form of tarsal conjunctival thickening of the eyelid in patients with neurofibromatosis type 1 (NF1), common ocular complications of which are Lisch nodules, choroidal nodules, and optic nerve glioma. A 46-year-old female patient was diagnosed with neurofibroma after biopsy and removal of 2 lumbar level intradural masses 15 years ago. She was being monitored without recurrence. When the patient visited our hospital, multiple iris Lisch nodules were found in both her eyes with ill-defined, diffuse thickening in the upper eyelid tarsal conjunctiva of the right eye. Neurofibroma was diagnosed by incisional biopsy and immunohistochemistry of the tarsal conjunctiva. The patient of the present case did not undergo any additional surgical treatment because tarsal conjunctiva thickening caused little functional problem. The patient has only been regularly examined for changes in size of neurofibroma, and there was no change in size over a 12-month period. Neurofibroma should be considered as a differential diagnosis if a patient diagnosed with NF1 shows tarsal conjunctiva thickening.'],\n",
       "       [26965370,\n",
       "        'In adult zebrafish, relatively quiescent progenitor cells show lesion-induced generation of motor neurons. Developmental motor neuron generation from the spinal motor neuron progenitor domain (pMN) sharply declines at 48\\u2005hours post-fertilisation (hpf). After that, mostly oligodendrocytes are generated from the same domain. We demonstrate here that within 48\\u2005h of a spinal lesion or specific genetic ablation of motor neurons at 72\\u2005hpf, the pMN domain reverts to motor neuron generation at the expense of oligodendrogenesis. By contrast, generation of dorsal Pax2-positive interneurons was not altered. Larval motor neuron regeneration can be boosted by dopaminergic drugs, similar to adult regeneration. We use larval lesions to show that pharmacological suppression of the cellular response of the innate immune system inhibits motor neuron regeneration. Hence, we have established a rapid larval regeneration paradigm. Either mechanical lesions or motor neuron ablation is sufficient to reveal a high degree of developmental flexibility of pMN progenitor cells. In addition, we show an important influence of the immune system on motor neuron regeneration from these progenitor cells.'],\n",
       "       [31366534,\n",
       "        'To identify the reason for age and gender differences in cancer risk. Age-standardized incidence rates for 17 cancer types were compared between genders in 50 populations. For each cancer type, the female/male rate ratio was listed in fixed order of population. Correlation coefficients were calculated between these lists in all pairwise combinations. For each population, the female/male rate ratio was listed in fixed order of cancer. Correlation coefficients were calculated between lists in all pairwise combinations. Only four pairwise combinations for cancer type gave a correlation coefficient greater than 0.700. For each population, the lowest correlation coefficient was 0.950. The reason for the differences in risk of cancer varies with each type of cancer, but remains fixed in all populations. It is suspected that species-specific genes control stem cell telomere dynamics in a fixed strategy at rates that vary among tissues and between genders.'],\n",
       "       [31366514,\n",
       "        'Different entities can be the cause of scalp neoplasia. In a phenotype with multiple cystic scalp lesions, the diagnosis must be made with particular caution because the appearance of apparently benign tumors does not necessarily correspond to the biological behaviour of the lesions. This case report describes diagnosis and therapy of a patient with multiple cystic tumors confined to the scalp. Diagnosis of benign lesions all over the scalp allowed an aesthetically pleasing surgical treatment result. Long-term follow-up control was offered to the patient because the histological diagnosis identified further small tumors of the same type as the large lesions, so further neoplasms are likely to develop.'],\n",
       "       [30511208,\n",
       "        \"Peste-des-petits-ruminants is a highly contagious and fatal disease of goats and sheep caused by non-segmented, negative strand RNA virus belonging to the Morbillivirus genus-Peste-des-petits-ruminants virus (PPRV) which is evolutionarily closely related to Rinderpest virus (RPV). The large protein 'L' of the members of this genus is a multifunctional catalytic protein, which transcribes and replicates the viral genomic RNA as well as possesses mRNA capping, methylation and polyadenylation activities; however, the detailed mechanism of mRNA capping by PPRV L protein has not been studied. We have found earlier that the L protein of RPV has RNA triphosphatase (RTPase), guanylyltransferase (GTase) and methyltransferase activities, and unlike vesicular stomatitis virus (VSV), follows the conventional pathway of mRNA capping. In the present work, using a 5'-end labelled viral RNA as substrate, we demonstrate that PPRV L protein has RTPase activity when present in the ribonucleoprotein complex of purified virus as well as recombinant L-P complex expressed in insect cells. Further, a minimal domain in the C-terminal region (aa1640-1840) of the L protein has been expressed in E. coli and shown to exhibit RTPase activity. The RTPase activity of PPRV L protein is metal-dependent and functions with a divalent cation, either magnesium or manganese. In addition, RTPase associated nucleotide triphosphatase activity (NTPase) of PPRV L protein is also demonstrated. This work provides the first detailed study of RTPase activity and identifies the RTPase domain of PPRV L protein.\"],\n",
       "       [31362555,\n",
       "        'There is good scientific rationale to support the use of growth factors to promote musculoskeletal tissue regeneration. However, the clinical effectiveness of platelet-rich plasma (PRP) and other blood-derived products has yet to be proven. Characterization and reporting of PRP preparation protocols utilized in clinical trials for the treatment of musculoskeletal disease is highly inconsistent, and the majority of studies do not provide sufficient information to allow the protocols to be reproduced. Furthermore, the reporting of blood-derived products in orthopaedics is limited by the multiple PRP classification systems available, which makes comparison of results between studies challenging. Several attempts have been made to characterize and classify PRP; however, no consensus has been reached, and there is lack of a comprehensive and validated classification. In this annotation, we outline existing systems used to classify preparations of PRP, highlighting their advantages and limitations. There remains a need for standardized universal nomenclature to describe biological therapies, as well as a comprehensive and reproducible classification system for autologous blood-derived products. Cite this article: Bone Joint J 2019;101-B:891-896.'],\n",
       "       [26845534,\n",
       "        'Adult stem cells or residential progenitor cells are critical to maintain the structure and function of adult tissues (homeostasis) throughout the lifetime of an individual. Mis-regulation of stem cell proliferation and differentiation often leads to diseases including cancer, however, how wildtype adult stem cells and cancer cells respond to cellular damages remains unclear. We find that in the adult Drosophila midgut, intestinal stem cells (ISCs), unlike tumor intestinal cells, are resistant to various cellular damages. Tumor intestinal cells, unlike wildtype ISCs, are easily eliminated by apoptosis. Further, their proliferation is inhibited upon autophagy induction, and autophagy-mediated tumor inhibition is independent of caspase-dependent apoptosis. Interestingly, inhibition of tumorigenesis by autophagy is likely through the sequestration and degradation of mitochondria, as compromising mitochondria activity in these tumor models mimics the induction of autophagy and increasing the production of mitochondria alleviates the tumor-suppression capacity of autophagy. Together, these data demonstrate that wildtype adult stem cells and tumor cells show dramatic differences in sensitivity to cellular damages, thus providing potential therapeutic implications targeting tumorigenesis.'],\n",
       "       [28916802,\n",
       "        'Hormones play essential roles during development and maintaining homeostasis in adult organisms, regulating a plethora of biological processes. Generally, hormones are secreted by glands and perform a systemic action. Here we show that Juvenile Hormones (JHs), insect sesquiterpenoids synthesized by the corpora allata, are also synthesized by the adult Drosophila gut. This local, gut specific JH activity, is synthesized by and acts on the intestinal stem cell and enteroblast populations, regulating their survival and cellular growth through the JH receptors Gce/Met and the coactivator Tai. Furthermore, we show that this local JH activity is important for damage response and is necessary for intestinal tumor growth driven by activating mutations in Wnt and EGFR/Ras pathways. Together, our results identify JHs as key hormonal regulators of gut homeostasis and open the possibility that analogous hormones may play a similar role in maintaining vertebrate adult intestinal stem cell population and sustaining tumor growth.'],\n",
       "       [31693685,\n",
       "        'Circadian rhythms allow animals to coordinate behavioral and physiological processes with respect to one another and to synchronize these processes to external environmental cycles. In most animals, circadian rhythms are produced by core clock neurons in the brain that generate and transmit time-of-day signals to downstream tissues, driving overt rhythms. The neuronal pathways controlling clock outputs, however, are not well understood. Furthermore, it is unclear how the central clock modulates multiple distinct circadian outputs. Identifying the cellular components and neuronal circuitry underlying circadian regulation is increasingly recognized as a critical step in the effort to address health pathologies linked to circadian disruption, including heart disease and metabolic disorders. Here, building on the conserved components of circadian and metabolic systems in mammals and Drosophila melanogaster, we used a recently developed feeding monitor to characterize the contribution to circadian feeding rhythms of two key neuronal populations in the Drosophila pars intercerebralis (PI), which is functionally homologous to the mammalian hypothalamus. We demonstrate that thermogenetic manipulations of PI neurons expressing the neuropeptide SIFamide (SIFa) as well as mutations of the SIFa gene degrade feeding:fasting rhythms. In contrast, manipulations of a nearby population of PI neurons that express the Drosophila insulin-like peptides (DILPs) affect total food consumption but leave feeding rhythms intact. The distinct contribution of these two PI cell populations to feeding is accompanied by vastly different neuronal connectivity as determined by trans-Tango synaptic mapping. These results for the first time identify a non-clock cell neuronal population in Drosophila that regulates feeding rhythms and furthermore demonstrate dissociable control of circadian and homeostatic aspects of feeding regulation by molecularly-defined neurons in a putative circadian output hub.'],\n",
       "       [31696229,\n",
       "        'Sigma-1 receptor (S1R) is an endoplasmic reticulum (ER) chaperone that not only regulates mitochondrial respiration but also controls cellular defense against ER and oxidative stress. This makes S1R a potential therapeutic target in amyotrophic lateral sclerosis (ALS). Especially, as a missense mutation E102Q in S1R has been reported in few familial ALS cases. However, the pathogenicity of S1RE102Q and the beneficial impact of S1R in the ALS context remain to be demonstrated in vivo. To address this, we generated transgenic Drosophila that expresses human wild-type S1R or S1RE102Q. Expression of mutant S1R in fly neurons induces abnormal eye morphology and locomotor defects in a dose-dependent manner. This was accompanied by abnormal mitochondrial fragmentation, reduced adenosine triphosphate (ATP) levels and a higher fatigability at the neuromuscular junction during high energy demand. Overexpressing IP3 receptor or glucose transporter mitigates the S1RE102Q-induced eye phenotype, further highlighting the role of calcium and energy metabolism in its toxicity. More importantly, we showed that wild-type S1R rescues locomotor activity and ATP levels of flies expressing the key ALS protein, TDP43. Moreover, overexpressing wild-type S1R enhances resistance of flies to oxidative stress. Therefore, our data provide the first genetic evidence that mutant S1R recapitulates ALS pathology in vivo while increasing S1R confers neuroprotection against TDP43 toxicity.'],\n",
       "       [31701127,\n",
       "        'CoREST has been identified as a subunit of several protein complexes that generate transcriptionally repressive chromatin structures during development. However, a comprehensive analysis of the CoREST interactome has not been carried out. We use proteomic approaches to define the interactomes of two dCoREST isoforms, dCoREST-L and dCoREST-M, in Drosophila. We identify three distinct histone deacetylase complexes built around a common dCoREST/dRPD3 core: A dLSD1/dCoREST complex, the LINT complex and a dG9a/dCoREST complex. The latter two complexes can incorporate both dCoREST isoforms. By contrast, the dLSD1/dCoREST complex exclusively assembles with the dCoREST-L isoform. Genome-wide studies show that the three dCoREST complexes associate with chromatin predominantly at promoters. Transcriptome analyses in S2 cells and testes reveal that different cell lineages utilize distinct dCoREST complexes to maintain cell-type-specific gene expression programmes: In macrophage-like S2 cells, LINT represses germ line-related genes whereas other dCoREST complexes are largely dispensable. By contrast, in testes, the dLSD1/dCoREST complex prevents transcription of germ line-inappropriate genes and is essential for spermatogenesis and fertility, whereas depletion of other dCoREST complexes has no effect. Our study uncovers three distinct dCoREST complexes that function in a lineage-restricted fashion to repress specific sets of genes thereby maintaining cell-type-specific gene expression programmes.'],\n",
       "       [27187149,\n",
       "        'Homeostatic renewal of many adult tissues requires balanced self-renewal and differentiation of local stem cells, but the underlying mechanisms are poorly understood. Here we identified a novel feedback mechanism in controlling intestinal regeneration and tumorigenesis in Drosophila. Sox21a, a group B Sox protein, is preferentially expressed in the committed progenitor named enteroblast (EB) to promote enterocyte differentiation. In Sox21a mutants, EBs do not divide, but cannot differentiate properly and have increased expression of mitogens, which then act as paracrine signals to promote intestinal stem cell (ISC) proliferation. This leads to a feedback amplification loop for rapid production of differentiation-defective EBs and tumorigenesis. Notably, in normal intestine following damage, Sox21a is temporally downregulated in EBs to allow the activation of the ISC-EB amplification loop for epithelial repair. We propose that executing a feedback amplification loop between stem cells and their progeny could be a common mechanism underlying tissue regeneration and tumorigenesis.'],\n",
       "       [31664682,\n",
       "        'Mutations in LRRK2 are currently recognized as the most common monogenetic cause of Parkinsonism. The elevation of kinase activity of LRRK2 that frequently accompanies its mutations is widely thought to contribute to its toxicity. Accordingly, many groups have developed LRRK2-specific kinase inhibitors as a potential therapeutic strategy. Given that protein phosphorylation is a reversible event, we sought to elucidate the phosphatase(s) that can reverse LRRK2-mediated phosphorylation, with the view that targeting this phosphatase(s) may similarly be beneficial. Using an unbiased RNAi phosphatase screen conducted in a Drosophila LRRK2 model, we identified PP2A as a genetic modulator of LRRK2-induced neurotoxicity. Further, we also identified ribosomal S6 kinase (S6K), a target of PP2A, as a novel regulator of LRRK2 function. Finally, we showed that modulation of PP2A or S6K activities ameliorates LRRK2-associated disease phenotype in Drosophila.'],\n",
       "       [31676440,\n",
       "        'The C. elegans lipase-like 5 (lipl-5) gene is predicted to code for a lipase homologous to the human gastric acid lipase. Its expression was previously shown to be modulated by nutritional or immune cues, but nothing is known about its impact on the lipid landscape and ensuing functional consequences. In the present work, we used mutants lacking LIPL-5 protein and found that lipl-5 is important for normal lipidome composition as well as its remodeling in response to food deprivation. Particularly, lipids with signaling functions such as ceramides and mitochondrial lipids were affected by lipl-5 silencing. In comparison with wild type worms, animals lacking LIPL-5 were enriched in cardiolipins linked to polyunsaturated C20 fatty acids and coenzyme Q-9. Differences in mitochondrial lipid composition were accompanied by differences in mitochondrial activity as mitochondria from well-fed lipl-5 mutants were significantly more able to oxidize respiratory substrates when compared with mitochondria from well-fed wild type worms. Strikingly, starvation elicited important changes in mitochondrial activity in wild type worms, but not in lipl-5 worms. This indicates that this lipase is a determinant of mitochondrial functional remodeling in response to food withdrawal.'],\n",
       "       [31695036,\n",
       "        'Charcot-Marie-Tooth disease (CMT) is a length-dependent peripheral neuropathy. The aminoacyl-tRNA synthetases constitute the largest protein family implicated in CMT. Aminoacyl-tRNA synthetases are predominantly cytoplasmic, but are also present in the nucleus. Here we show that a nuclear function of tyrosyl-tRNA synthetase (TyrRS) is implicated in a Drosophila model of CMT. CMT-causing mutations in TyrRS induce unique conformational changes, which confer capacity for aberrant interactions with transcriptional regulators in the nucleus, leading to transcription factor E2F1 hyperactivation. Using neuronal tissues, we reveal a broad transcriptional regulation network associated with wild-type TyrRS expression, which is disturbed when a CMT-mutant is expressed. Pharmacological inhibition of TyrRS nuclear entry with embelin reduces, whereas genetic nuclear exclusion of mutant TyrRS prevents hallmark phenotypes of CMT in the Drosophila model. These data highlight that this translation factor may contribute to transcriptional regulation in neurons, and suggest a therapeutic strategy for CMT.'],\n",
       "       [31652654,\n",
       "        'Regenerative retinal therapies have introduced progenitor cells to replace dysfunctional or injured neurons and regain visual function. While contemporary cell replacement therapies have delivered retinal progenitor cells (RPCs) within customized biomaterials to promote viability and enable transplantation, outcomes have been severely limited by the misdirected and/or insufficient migration of transplanted cells. RPCs must achieve appropriate spatial and functional positioning in host retina, collectively, to restore vision, whereas movement of clustered cells differs substantially from the single cell migration studied in classical chemotaxis models. Defining how RPCs interact with each other, neighboring cell types and surrounding extracellular matrixes are critical to our understanding of retinogenesis and the development of effective, cell-based approaches to retinal replacement. The current article describes a new bio-engineering approach to investigate the migratory responses of innate collections of RPCs upon extracellular substrates by combining microfluidics with the well-established invertebrate model of Drosophila melanogaster. Experiments utilized microfluidics to investigate how the composition, size, and adhesion of RPC clusters on defined extracellular substrates affected migration to exogenous chemotactic signaling. Results demonstrated that retinal cluster size and composition influenced RPC clustering upon extracellular substrates of concanavalin (Con-A), Laminin (LM), and poly-L-lysine (PLL), and that RPC cluster size greatly altered collective migratory responses to signaling from Fibroblast Growth Factor (FGF), a primary chemotactic agent in Drosophila. These results highlight the significance of examining collective cell-biomaterial interactions on bio-substrates of emerging biomaterials to aid directional migration of transplanted cells. Our approach further introduces the benefits of pairing genetically controlled models with experimentally controlled microenvironments to advance cell replacement therapies.'],\n",
       "       [31389510,\n",
       "        'The objective of this study was to investigate the effects of low triiodothyronine syndrome (LT3S) on platelet function and clotting factors in patients with nephrotic syndrome(NS). Patients with primary nephrotic syndrome were divided into two groups, normal thyroid function (group A) and LT3S (group B), based on whether they had LT3S or not. Healthy subjects were selected as the control group (group C). Blood coagulation function was detected in each group. The platelet activation function (CD62P, CD63) was determined by flow cytometry. The platelet aggregation rate was detected by an optical method using adenosine diphosphate and arachidonic acid as inducers. The proportion of primary nephrotic syndrome with LT3S was 23.2% (69/298). Compared with group C, group A had higher CD62P and PAgTADP, and group B had higher CD62P, CD63, PAgTAA, and PAgTADP; the difference was statistically significant (all P < 0.05). There was no significant difference in renal pathology between group A and group B (X2 = 4.957, P = 0.421). Compared with group A, the 24-hour urine protein, CD63, PAgTAA, and PAgTADP were higher in group B, and APTT and Alb were lower. The difference was statistically significant (P < 0.05). Logistic regression analysis showed that LT3S was associated with CD36 (OR: 3.516; 95% CI: 1.742~8.186; P = 0.004) and PAgTAA (OR: 0.442; 95% CI: 1.001~1.251; P = 0.037). NS patients are prone to LT3S. Patients with LT3S may have abnormal platelet activation and increase of platelet aggregation.'],\n",
       "       [31326549,\n",
       "        'MYC deregulation contributes to breast cancer development and progression. Deregulated expression levels of long non-coding RNAs (lncRNA) have been demonstrated to be critical players in development and/or maintenance of breast cancer. In this study we aimed to evaluate lncRNA expressions depending on MYC overexpression and knockdown in breast cancer cells. Cells were infected with lentiviral vectors by either knockdown or overexpression of c-MYC. LncRNA cDNA was transcribed from total RNA samples and lncRNAs were evaluated by qRT-PCR. Our results indicated that some of the lncRNAs having tumor suppressor (GAS5, MEG3, lincRNA-p21) and oncogenic roles (HOTAIR) are regulated by c-MYC. We observed that c-MYC regulates lncRNAs that have important roles on proliferation, cell cycle and etc. Further studies will give us a light to identify molecular mechanisms related to MYC-lncRNA regulatory pathways in breast cancer.'],\n",
       "       [31221976,\n",
       "        'Human cytomegalovirus (CMV) causes a wide array of disease to diverse populations of immune-compromised individuals. Thus, a more comprehensive understanding of how CMV enters numerous host cell types is necessary to further delineate the complex nature of CMV pathogenesis and to develop targeted therapeutics. To that end, we establish a vaccination strategy utilizing membrane vesicles derived from epithelial cells to generate a library of monoclonal antibodies (mAbs) targeting cell surface proteins in their native conformation. A high-throughput inhibition assay is employed to screen these antibodies for their ability to limit infection, and mAbs targeting CD46 are identified. In addition, a significant reduction of viral proliferation in CD46-KO epithelial cells confirms a role for CD46 function in viral dissemination. Further, we demonstrate a CD46-dependent entry pathway of virus infection in trophoblasts, but not in fibroblasts, highlighting the complexity of CMV entry and identifying CD46 as an entry factor in congenital infection.'],\n",
       "       [31348317,\n",
       "        \"Gastric neuroendocrine carcinoma (NEC) is rare. It is considered to be aggressive and has a poor prognosis since the diagnosis is usually made at its advanced stage. However, the survival rate is increased in some early gastric NECs. This study showed a case and reviewed the clinical characteristics of early NECs in stomach. A 38-year-old man displayed no symptoms and underwent the gastric endoscopy test for his health examination, which showed a red slightly depressed lesion 1.0\\u200acm in size on the lesser curvature of gastric cardia. Magnifying endoscopy with narrow-band imaging (NBI) revealed a clear demarcation and an irregular mesh in vessels within the depressed area. The background mucosa was negative for atrophic gastritis and Helicobacter Pylori infection. A contrast-enhanced computed tomography (CT) scan disclosed no obvious thickening of stomach and lymphadenopathy. Blood tests and physical examination were unremarkable. He had not received any surgical treatment and denied a family history of cancer and any genetic disorders. The pathologic result of biopsy from the lesion was suspicious of superficial carcinoma. Then endoscopic submucosal dissection (ESD) was performed. Gastric NEC G3 in the early stage (T1aN0M0). Concerning this patient's situation, we considered the ESD as a curable treatment. And no radical surgery or adjuvant chemotherapy was arranged. The patient is doing well and displays no recurrence for 11\\u200amonths, who is still in follow-up. The early diagnosis and effective treatment by endoscopy would contribute to improve the prognosis of gastric NECs.\"],\n",
       "       [31221067,\n",
       "        \"Generation of patient-specific induced pluripotent stem cells (iPSCs) and their subsequent differentiation into cardiomyocytes opened new opportunities for studying pathogenesis of inherited cardiovascular diseases. One of these diseases is hypertrophic cardiomyopathy (HCM) for which no efficient therapy methods have been developed so far. In this study, the approach based on patient-specific iPSCs was applied to create a model of the disease. Genetic analysis of a hypertrophic cardiomyopathy patient revealed R326Q mutation in the MYBPC3 gene. iPSCs of the patient were generated and characterized. The cells were differentiated into cardiomyocytes together with the control iPSCs from a healthy donor. The patient's iPSC-derived cardiomyocytes exhibited early HCM features, such as abnormal calcium handling and increased intracellular calcium concentration. Therefore, cardiomyocytes obtained by directed differentiation of iPSCs from the HCM patient can be used as a model system to study HCM pathogenesis.\"],\n",
       "       [31366500,\n",
       "        'Signaling regulation of myeloid zinc finger 1 (MZF1) has been implicated in the progression of many human malignancies; however, the mechanistic action of MZF1 in triple-negative breast cancer (TNBC) progression remains elusive. In this study, the aim was to investigate the molecular mechanisms of MZF1 and its functional role in TNBC cellular migration and invasion. Hs578T and MDA-MB-231 cells were transfected to stably express the acidic domain of MZF1 (MZF160-72), or were transfected with MZF1-specific or ELK1-specific short hairpin RNA (shRNA). Changes in cell morphology and distributions of cellular proteins were observed and subsequently migration and invasion were measured by wound healing and transwell assays. Expression levels of epithelial-mesenchymal transition (EMT)-related genes were carried out using immunoblotting and quantitative reverse transcription-polymerase chain reaction (RT-PCR) assays. Data of transcriptional regulation were obtained from promoter-luciferase reporter and chromatin immunoprecipitation (ChIP) assays. Herein, we found that MZF1 in high-level MZF1-expressing TNBC cells is associated with cell migration, invasion, and mesenchymal phenotype. MZF1 interacted with the promoter region of insulin-like growth factor 1 receptor (IGF1R) to drive invasion and metastasis of high-level MZF1-expressing TNBC cells. Exogenous expression of the acidic domain of MZF1 repressed the binding of endogenous MZF1 to IGF1R promoter via blocking the interaction with ETS-like gene 1 (ELK1). This blockage not only caused MZF1 protein degradation, but also restrained ELK1 nuclear localization in high-level MZF1-expressing TNBC cells. MZF1, but not ELK1, was necessary for the retention of mesenchymal phenotype by repressing IGF1R promoter activity in TNBC cells expressing high levels of MZF1. Activation of the IGF1R-driven p38MAPK-ERα-slug-E-cadherin signaling axis mediated the conversion of mesenchymal cell to epithelial phenotype, caused by MZF1 destabilization. These results suggest that MZF1 is an oncogenic inducer. Blocking of the MZF1/ELK1 interaction to reduce MZF1 protein stability by saturating the endogenous MZF1/ELK1 binding domains might be a promising therapeutic strategy for the treatment of high-level MZF1-expressing TNBC.'],\n",
       "       [31209254,\n",
       "        'Ran is a nucleocytoplasmic shuttle protein that is involved in cell cycle regulation, nuclear-cytoplasmic transport, and cell transformation. Ran plays an important role in cancer cell survival and cancer progression. Here, we show that, in addition to the nucleocytoplasmic localization of Ran, this GTPase is specifically associated with the plasma membrane/ruffles of ovarian cancer cells. Ran depletion has a drastic effect on RhoA stability and inhibits RhoA localization to the plasma membrane/ruffles and RhoA activity. We further demonstrate that the DEDDDL domain of Ran is required for the interaction with serine 188 of RhoA, which prevents RhoA degradation by the proteasome pathway. Moreover, the knockdown of Ran leads to a reduction of ovarian cancer cell invasion by impairing RhoA signalling. Our findings provide advanced insights into the mode of action of the Ran-RhoA signalling axis and may represent a potential therapeutic avenue for drug development to prevent ovarian tumour metastasis.'],\n",
       "       [31213813,\n",
       "        'Purpose: As well as functioning as a ligand that is selectively internalized by cells overexpressing human epidermal growth factor receptor-2 (HER2), HApt can exert cytotoxic effects by inducing cross-linking and subsequent translocation of HER2 to cytoplasmic vesicles, such downregulation of HER2 inhibits cell proliferation and induces apoptosis. We aimed to exploit the potential of HApt as both a targeting agent and antagonist to maximize the efficacy of mesoporous silica nanoparticle (MSN)-based drug release systems for HER2-positive breast cancer. Materials and methods: We fabricated novel HApt aptamer-functionalized pH-sensitive β-cyclodextrin (β-CD)-capped doxorubicin (DOX)-loaded mesoporous silica nanoparticles (termed MSN-BM/CD-HApt@DOX) for targeted delivery and selective targeting of HER2-positive cells. MSN-functionalized benzimidazole (MSN-BM) was used to load and achieve pH stimuli-responsive release of the chemotherapeutic agent doxorubicin (DOX). β-cyclodextrin was introduced as a gatekeeper for encapsulated DOX and HApt as a selective HER2-targeting moiety and biotherapeutic agent. Results: Physical and chemical characterizations (FT-IR, XRD, TEM and BET) confirmed successful construction of MSN-BM/CD-HApt@DOX nanoparticles. In vitro release assays verified pH-sensitive DOX release. MSN-BM/CD-HApt@DOX (relative DOX concentration, 3.6\\xa0μg/mL) underwent HER2-mediated endocytosis and was more cytotoxic to HER2-positive SKBR3 cells than HER2-negative MCF7 cells. MSN-BM/CD-HApt@DOX also exhibited better uptake and stronger growth inhibition in SKBR3 cells than the control MSN-BM/CD-NCApt@DOX functionalized with a scrambled nucleotide sequence on CD. Overall, intracellular delivery of DOX and the biotherapeutic agent HApt resulted in synergistic cytotoxic effects in HER2-positive cancer cells in comparison to either DOX or HApt alone. Conclusion: MSN-BM/CD-HApt@DOX enables HER2-mediated targeting and biotherapeutic effects as well as pH-responsive DOX drug release, resulting in synergistic cytotoxic effects in HER2-overexpressing cells in vitro. This novel nanocarrier could potentially enable specific targeting to improve the efficacy of chemotherapy for HER2-positive cancer.'],\n",
       "       [31262867,\n",
       "        'DOG1 is a calcium-activated chloride channel that has gained attention as a promising drug target due to its involvement in several processes essential for tumor development and progression. DOG1 is overexpressed in >95% of gastrointestinal stromal tumors (GIST). The aim was to determine DOG1 inhibition antitumoral effects on GIST. Human GIST (GIST-T1 and GIST882) cell lines were used to study the effect of DOG1 inhibitors on chloride currents, viability, colony formation, and cell cycle. CaCCinh-A01 decreased chloride currents. CaCCinh-A01 and T16inh-A01 reduced GIST cell viability and CaCCinh-A01 affected cell cycle distribution leading to G1 cell-cycle arrest. CaCCinh-A01 also increased the sub-G1 phase population, indicative of apoptosis, in GIST882. CaCCinh-A01 strongly reduced the colony forming ability of the cells, whereas T16inh-A01 did not. DOG1 inhibition has antitumoral effects in GIST cells in vitro, and could potentially serve as a target for GIST therapy.'],\n",
       "       [31226288,\n",
       "        'Liver cancer is one of the most frequently occurring types of cancer with high mortality rate. Hepatocellular carcinoma (HCC) frequently metastasizes to lung, portal vein, and portal lymph nodes and most HCCs show strong resistance to conventional anticancer drugs. Cancer stem cells (CSCs) are considered to be responsible for resistance to therapies. Hence, recent advancements in the use of liver cancer stem cells (LCSCs) are rapidly gaining recognition as an efficient and organized means for developing antitumor agents. We aimed to use a non-target-based high-throughput screening (HTS) approach to specifically target α-fetoprotein (AFP)+/cluster of differentiation (CD)133+ HCC present in mixed populations of HCC cells and hepatocytes. Herein, we identified actinomycin D (ActD) as a potential antitumor agent that significantly inhibits activity of LCSCs without affecting the co-cultured hepatocytes. To determine the mechanism of ActD-induced tumor-specificity in LCSC, we applied various cell-based assay models in vitro. In fact, ActD significantly increased reactive oxygen species (ROS) accumulation and DNA damage in Huh7 HCC cells, but not in Fa2N-4\\u202fcells, immortalized hepatocytes. Treatment of spheroid-forming LCSCs with ActD effectively decreased spheroid formation and the CD133+ HCC cell population. Importantly, these ActD-mediated effects are a result of inhibition of cystine/glutamate transporter xCT expression, via attenuation of CD133 synthesis. These results indicate that ActD suppresses stemness and malignant properties in HCC cells through destabilization of xCT, by inhibition of CD133 expression in LCSCs. The effects of ActD on LCSCs provide novel therapeutic strategies for targeting cancer stem-like cells in liver cancer.'],\n",
       "       [30650584,\n",
       "        'The aim of this study was to elucidate the impact of autologous umbilical cord blood cells (USSC) on bone regeneration and biomechanical stability in an ovine tibial bone defect. Ovine USSC were harvested and characterized. After 12 months, full-size 2.0 cm mid-diaphyseal bone defects were created and stabilized by an external fixateur containing a rigidity measuring device. Defects were filled with (i) autologous USSC on hydroxyapatite (HA) scaffold (test group), (ii) HA scaffold without cells (HA group), or (iii) left empty (control group). Biomechanical measures, standardized X-rays, and systemic response controls were performed regularly. After six months, bone regeneration was evaluated histomorphometrically and labeled USSC were tracked. In all groups, the torsion distance decreased over time, and radiographies showed comparable bone regeneration. The area of newly formed bone was 82.5 ± 5.5% in the control compared to 59.2 ± 13.0% in the test and 48.6 ± 2.9% in the HA group. Labeled cells could be detected in lymph nodes, liver and pancreas without any signs of tumor formation. Although biomechanical stability was reached earliest in the test group with autologous USSC on HA scaffold, the density of newly formed bone was superior in the control group without any bovine HA.'],\n",
       "       [31116257,\n",
       "        'Excessive pro-inflammatory cytokines result in adverse pregnancy outcomes, including preeclampsia-like phenotypes, and fetal growth restriction. Anti-inflammation might be an effective therapy. The aim of this research was to investigate whether Uncaria rhynchophylla alkaloid extract (URE), a highly safe anti-inflammation constituent of the herb, can inhibit inflammation and improve clinical characteristics of preeclampsia in a lipopolysaccharide (LPS)-induced preeclampsia rat model. The rat model was established by daily administration of LPS (1 μg/kg body weight per day) from gestational day (GD) 14 to 19. Different doses of URE (35, 70, and 140 mg/kg body weight per day) were administered from GD 14 to GD 19. The effects of URE on proteinuria, maternal hypertension, pregnancy outcomes, as well as pro-inflammatory cytokines levels in serum and placenta were measured. High-dose URE (HURE) treatment decreased LPS-induced mean 24-h proteinuria and systolic blood pressure, and increased fetal weight, placental weight, and the number of live pups (P<0.05). Moreover, increased serum and placental levels of interleukin (IL)-6, IL-1β, tumor necrosis factor-α, and interferon-γ in the LPS-treated group were obviously inhibited after HURE administration (P<0.01). URE improved preeclampsia symptoms and mitigated inflammatory responses in the LPS-induced preeclampsia rat model, which suggests that the anti-inflammation effect of URE might be an alternative therapy for preeclampsia.'],\n",
       "       [31166414,\n",
       "        'Infection, inflammation and bone resorption are closely related events in apical periodontitis development. Therefore, we sought to investigate the role of cyclooxygenase (COX) in osteoclastogenesis and bone metabolism signaling in periapical bone tissue after bacterial lipopolysaccharide (LPS) inoculation into root canals. Seventy two C57BL/6 mice had the root canals of the first molars inoculated with a solution containing LPS from E. coli (1.0 mg/mL) and received selective (celecoxib) or non-selective (indomethacin) COX-2 inhibitor. After 7, 14, 21 and 28 days the animals were euthanized and the tissues removed for total RNA extraction. Evaluation of gene expression was performed by qRT-PCR. Statistical analysis was performed using analysis of variance (ANOVA) followed by post-tests (α=0.05). LPS induced expression of mRNA for COX-2 (Ptgs2) and PGE2 receptors (Ptger1, Ptger3 and Ptger4), indicating that cyclooxygenase is involved in periapical response to LPS. A signaling that favours bone resorption was observed because Tnfsf11 (RANKL), Vegfa, Ctsk, Mmp9, Cd36, Icam, Vcam1, Nfkb1 and Sox9 were upregulated in response to LPS. Indomethacin and celecoxib differentially modulated expression of osteoclastogenic and other bone metabolism genes: celecoxib downregulated Igf1r, Ctsk, Mmp9, Cd36, Icam1, Nfkb1, Smad3, Sox9, Csf3, Vcam1 and Itga3 whereas indomethacin inhibited Tgfbr1, Igf1r, Ctsk, Mmp9, Sox9, Cd36 and Icam1. We demonstrated that gene expression for COX-2 and PGE2 receptors was upregulated after LPS inoculation into the root canals. Additionally, early administration of indomethacin and celecoxib (NSAIDs) inhibited osteoclastogenic signaling. The relevance of the cyclooxygenase pathway in apical periodontitis was shown by a wide modulation in the expression of genes involved in both bone catabolism and anabolism.'],\n",
       "       [23242220,\n",
       "        'OBJECTIVE, MATERIAL AND METHODS: The aim of this study was to analyse the protein fractions of the soluble parasitic antigen (SPA) from in vitro cultures of the native Polish strains of Babesia canis canis and to determine their immunogenicity through Western blotting using the sera of dogs vaccinated with this antigen. Polyacrylamide gel electrophoresis revealed 21 protein fractions with molecular weights from 12 to 205 kDa. The most intense reaction in Western blotting was observed between the serum antibodies of the SPA-vaccinated dogs and the fraction with the molecular weight of 52 kDa. Detailed studies on the composition of SPA of Babesia canis canis and reactivity of its individual protein fractions can be a starting point for the development of subunit vaccines against babesiosis. Using a preparation with only some electrophoretic fractions of SPA in the production of vaccines would allow to avoid putting an unnecessary protein burden in the vaccine which could cause side effects.'],\n",
       "       [31139925,\n",
       "        'Advanced oral squamous cell carcinomas (OSCC) have limited therapeutic options. Although immune therapies are emerging as a potentially effective alternative or adjunct to chemotherapies, the therapeutic efficacy of combination immune chemotherapies has yet to be determined. Using a 4-nitroquinolone-N-oxide (4NQO) orthotopic model of OSCC in immunocompetent mice, we evaluated the therapeutic efficacy of single- and combined-agent treatment with a poly-epitope tumor peptide vaccine, cisplatin and/or an A2AR inhibitor, ZM241385. The monotherapies or their combinations resulted in a partial inhibition of tumor growth and, in some cases, a significant but transient upregulation of systemic anti-tumor CD8+ T cell responses. These responses eroded in the face of expanding immunoregulatory cell populations at later stages of tumor progression. Our findings support the need for the further development of combinatorial therapeutic approaches that could more effectively silence dominant immune inhibitory pathways operating in OSCC and provide novel, more beneficial treatment options for this tumor.'],\n",
       "       [29735702,\n",
       "        'Expression of HMGA2 is strongly associated with body size and growth in mice and humans. In mice, inactivation of one or both alleles of Hmga2 results in body-size reductions of 20% and 60%, respectively. In humans, microdeletions involving the HMGA2 locus result in short stature, suggesting the function of the HMGA2 protein is conserved among mammals. To test this hypothesis, we generated HMGA2-deficient pigs via gene editing and somatic cell nuclear transfer (SCNT). Examination of growth parameters revealed that HMGA2-/+ male and female pigs were on average 20% lighter and smaller than HMGA2+/+ matched controls (P < 0.05). HMGA2-/- boars showed significant size reduction ranging from 35 to 85% of controls depending on age (P < 0.05), and organ weights were also affected (P < 0.05). HMGA2-/+ gilts and boars exhibited normal reproductive development and fertility, while HMGA2-/- boars were sterile due to undescended testes (cryptorchidism). Crossbreeding HMGA2-/+ boars and gilts produced litters lacking the HMGA2-/- genotype. However, analysis of day (D) D40 and D78 pregnancies indicated that HMGA2-/- fetuses were present at the expected Mendelian ratio, but placental abnormalities were seen in the D78 HMGA2-/- concepti. Additionally, HMGA2-/- embryos generated by gene editing and SCNT produced multiple pregnancies and viable offspring, indicating that lack of HMGA2 is not lethal per se. Overall, our results show that the effect of HMGA2 with respect to growth regulation is highly conserved among mammals and opens up the possibility of regulating body and organ size in a variety of mammalian species including food and companion animals.'],\n",
       "       [31124490,\n",
       "        'To evaluate the effect of cyanidin-3-glucoside (C3G) in oxygen-induced retinopathy (OIR) mouse model. In this experimental study, 10 C57BL / 6J type mice exposed to room air comprised two control groups (n = 5 each; a negative control and a group receiving intravitreal sterile dimethyl sulfoxide [IVS DMSO]). Thirty C57BL / 6J type mice exposed to 75% ± 2% oxygen from postnatal day 7 to postnatal day 12 comprised the OIR groups. On postnatal day 12, these mice were randomized into six groups (n = 5 each): two OIR control groups (negative control and IVS DMSO), two intravitreal C3G groups (300 and 600 ng/μL), and two intraperitoneal C3G groups (0.05 and 0.1 mg/kg). We quantified neovascularization by counting endothelial cell proliferation on the vitreal side of the inner limiting membrane of the retina and examined histological and ultrastructural changes via light and electron microscopy and apoptosis by terminal deoxynucleotidyl transferase deoxy-UTP-nick end labeling. The intravitreal C3G groups yielded lower endothelial cell counts compared with the intravitreal DMSO group. The intraperitoneal high-dose group had lower cell counts compared with the OIR control groups. Electron microscopy revealed significantly less mitochondrial dysmorphology in intravitreal groups and the high-dose intraperitoneal mice. We noted no difference in apoptotic cell count between the controls, low-dose intravitreal, and both intraperitoneal groups. However, apoptotic cell count was significantly higher in the high-dose intravitreal group. C3G suppresses endothelial cell proliferation in an OIR mouse model, leads to a reduced hyperoxia-induced mitochondrial dysmorphology, but increases apoptotic cell death in high concentrations.'],\n",
       "       [29678567,\n",
       "        'Mammalian testes are maintained at a relatively lesser temperature than the abdominal region so that normal spermatogenesis can occur. Germ cell apoptosis has resulted in heat-damaged testes that occurs as a result of cryptorchidism, but the mechanism is not yet fully understood. To elucidate the cause of germ-cell death by cryptorchidism, cryptorchidism was surgically induced in dog testes and histological and molecular analyses were performed. Histological data indicated that the seminiferous tubules of cryptorchid testes and epididymis contained fewer germ cells. Total RNA sequencing was performed to screen for overexpressed genes in cryptorchid dog testes. Clusterin RNA was in greater abundance (approximately 12.8-fold) in cryptorchid testes than in normal testes. In addition, cleaved caspase-3 and -8 were detected in greater abundance in cryptorchid dog testes. Real time RT-PCR and western blotting analysis indicated there was a greater abundance of clusterin in cryptorchid dog testes. Furthermore, clusterin was detected in extracellular regions of cryptorchid dog testes during the 4 weeks after surgery. Thus, germ-cell specific apoptosis and expression of clusterin genes occur with a resulting presence of this protein in extracellular regions of cryptorchid dog testes. This result will facilitate further study of spermatogenesis and the specific mechanisms by which cryptorchidism results in male infertility.'],\n",
       "       [29378029,\n",
       "        'Porcine epidemic diarrhea virus (PEDV) infects enterocytes and in nursery pigs, results in diarrhea, anorexia, and reduced performance. Therefore, the objective of this study was to determine how PEDV infection influenced growth performance and repartitioning of amino acids and energy in nursery pigs. A total of 32 barrows and gilts, approximately 1 wk post-wean (BW = 8.46 ± 0.50 kg), and naïve for PEDV were obtained, weighed, and allotted based on sex and BW to one of two treatments: 1) Control, PEDV naïve and 2) PEDV-inoculated (PEDV) with eight pens of two pigs each per treatment. On day post-inoculation (dpi) 0, PEDV pigs were inoculated via intragastric gavage with PEDV isolate (USA/Iowa/18984/2013). Pig and feeder weights were recorded at dpi -7, 0, 5, and 20 in order to calculate ADG, ADFI, and G:F. Eight pigs per treatment were euthanized on dpi 5 and 20, and tissues and blood were collected. At dpi 5, all PEDV pigs were PCR positive for PEDV in feces. Overall, PEDV pigs tended (P < 0.10) to increase ADFI, which resulted in reduced (P < 0.05) feed efficiency. At dpi 5, PEDV pigs had reduced (P < 0.05) villus height and increased (P < 0.05) stem cell proliferation in the jejunum compared with Control pigs. Pigs inoculated with PEDV had increased (P < 0.05) serum haptoglobin and increased insulin-to-glucose ratios compared with Control pigs at dpi 5. Markers of muscle proteolysis were not different (P > 0.05) between treatments within dpi; however, at dpi 5, 20S proteasome activity was increased (P < 0.05) in longissimus dorsi of PEDV pigs compared with Control pigs. Liver and jejunum gluconeogenic enzyme activities were not different (P > 0.05) between treatments within dpi. Overall, PEDV-inoculated pigs did recover the absorptive capacity that was reduced during PEDV infection by increasing proliferation of intestinal stem cells. However, the energy and nutrients needed to recover the epithelium may be originating from available luminal nutrients instead of muscle proteolysis and gluconeogenesis. This study provides insight into the effects of an enteric coronavirus on postabsorptive metabolism in nursery pigs.'],\n",
       "       [27791406,\n",
       "        \"The attempt to limit the negative effects of polyester implants on the articular cavity by using preparations containing growth factors. Polyester implants used for the reconstruction of a rabbit's cranial cruciate ligament (CCL) were saturated with autogenic platelet-rich plasma (PRP), antlerogenic stem cells MIC-1 and their homogenate prior to the surgery. Six months after CCL reconstruction, morphological, and biochemical blood tests were carried out, including proteinogram and acute phase proteins. The knee joints were also examined macro- and microscopically. The results, compared to the control group, showed a favorable effect of the PRP and homogenate of antlerogenic cells on limiting the inflammation caused by the presence of polyester implant in the knee joint. The addition of growth factors caused covering the implant faster with the recipient's connective tissue, thus contributing to reducing the inflammatory reaction of the articular capsule to the presence of polyester. At the same time, no enhanced local or general reaction of the rabbit organism was observed to the presence of xenogenic antlerogenic stem cells MIC-1 homogenate which, like the PRP, may provide an easily available source of growth factors, increasingly often used in regenerative medicine. Applying antlerogenic stem cells, their homogenate or PRP increases the volume of connective tissue that surrounds and intertwines polyester CCL implant, separating it from synovial cavity environment.\"],\n",
       "       [30074186,\n",
       "        'This study was designed to evaluate the effects of drilling through the growth plate and using adipose-derived stem cells (ADSCs) and bone morphogenetic protein-2 (BMP-2) to treat femoral head epiphyseal ischemic necrosis, which can be done in juvenile rabbits. Passagefour bromodeoxyuridine (BrdU)-labeled ADSCs were cultured, assayed with MTT to determine their viability and stained with alizarin red dye to determine their osteogenic ability. Two-month-old, healthy male rabbits (1.2 to 1.4 kg, n=45) underwent ischemic induction and were randomly divided into five groups (group A: animal model control; group B: drilling; group C: drilling & ADSCs; group D: drilling & BMP-2; and group E: drilling & ADSCs & BMP-2). Magnetic resonance imaging (MRI), X-ray imaging, hematoxylin and eosin staining and BrdU immunofluorescence detection were applied 4, 6 and 10 weeks after treatment. Approximately 90% of the ADSCs were labeled with BrdU and showed good viability and osteogenic ability. Similar results were observed in the rabbits in groups C and E at weeks 6 and 10. The animals of groups C and E demonstrated normal hip structure and improved femoral epiphyseal quotients and trabecular areas compared with those of the groups A and B (P<0.01). Group D demonstrated improved femoral epiphyseal quotients and trabecular areas compared with those of groups A and B (P<0.05). In summary, drilling through the growth plate combined with ADSC and BMP-2 treatments induced new bone formation and protected the femoral head epiphysis from collapsing in a juvenile rabbit model of femoral head epiphyseal ischemic necrosis.'],\n",
       "       [31150628,\n",
       "        'The dose-dependent neuroprotective role of licorice-derived glycyrrhizin during subacute neuroterminal norepinephrine (NE) depletion was studied in rat brain. Experimental design included thirty 5-week-old male rats randomly divided into five groups. Compared to the saline-injected control group, the group receiving daily intraperitoneal injection of fusaric acid (FA; 5\\u202fmg/kg/b.w.) for 30 days showed pharmacological depletion of NE. The neuroprotective effects of three successively increasing oral doses of glycyrrhizin were examined in FA-treated rats. Neurochemical parameters and histo-/immunohistopathological changes in the hippocampus were examined. FA generated global hippocampal stress with altered neurobiochemical parameters, accompanied by immune-confirmed inflammatory tissue damage, and noticeable behavioral changes. Although glycyrrhizin after FA-induced intoxication did not correct the recorded drop in the NE level, it decreased the dopamine levels to control levels. Similarly, glycyrrhizin at a high dose restored the serotonin level to its normal value and blocked the FA-induced increase in the level of its metabolite, 5-hydroxyindoleacetic acid. The FA-induced rise in γ-aminobutyric acid (GABA) and histamine was alleviated after administration of a high dose of glycyrrhizin. This was accompanied by improvements in the bioenergetic status and neuronal regenerative capacity through recovery of ATP and brain-derived neurotrophic factor levels to the pre-intoxicated values. High doses of glycyrrhizin also ameliorated the FA-generated behavioral changes and oxidative damage, manifested by the reduction in the expression of cortical pro-apoptotic caspase 3 in the same group. This study suggests that glycyrrhizin can potentially mend most of the previously evoked neuronal damage induced by FA intoxication in the brain of an experimental rat model.'],\n",
       "       [31104401,\n",
       "        'Both Plasmodium spp. and Toxoplasma gondii are important apicomplexan parasites, which infect humans worldwide. Genetic analyses have revealed that 33% of amino acid sequences of inner membrane complex from the malaria parasite Plasmodium berghei is similar to that of Toxoplasma gondii. Inner membrane complex is known to be involved in cell invasion and replication. In this study, we investigated the resistance against T. gondii (ME49) infection induced by previously infected P. berghei (ANKA) in mice. Levels of T. gondii-specific IgG, IgG1, IgG2a, and IgG2b antibody responses, CD4+ and CD8+ T cell populations were found higher in the mice infected with P. berghei (ANKA) and challenged with T. gondii (ME49) compared to that in control mice infected with T. gondii alone (ME49). P. berghei (ANKA) + T. gondii (ME49) group showed significantly reduced the number and size of T. gondii (ME49) cysts in the brains of mice, resulting in lower body weight loss compared to ME49 control group. These results indicate that previous exposure to P. berghei (ANKA) induce resistance to subsequent T. gondii (ME49) infection.'],\n",
       "       [29390112,\n",
       "        'Most animals display retarded growth in adverse conditions; however, upon the removal of unfavorable factors, they often show quick growth restoration, which is known as \"catch-up\" growth. In zebrafish embryos, hypoxia causes growth arrest, but subsequent reoxygenation induces catch-up growth. Here, we report the role of insulin receptor substrate (Irs)1-mediated insulin/insulinlike growth factor signaling (IIS) and the involvement of stem cells in catch-up growth in reoxygenated zebrafish embryos. Disturbed irs1 expression attenuated IIS, resulting in greater inhibition in catch-up growth than in normal growth and forced IIS activation‒restored catch-up growth. The irs1 knockdown induced noticeable cell death in neural crest cells (NCCs; multipotent stem cells) under hypoxia, and the pharmacological/genetic ablation of NCCs hindered catch-up growth. Furthermore, inhibition of the apoptotic pathway by pan-caspase inhibition or forced activation of Akt signaling in irs1 knocked-down embryos blocked NCC cell death and rescued catch-up growth. Our data indicate that this multipotent stem cell is indispensable for embryonic catch-up growth and that Irs1-mediated IIS is a prerequisite for its survival under severe adverse environments such as prolonged hypoxia.'],\n",
       "       [29381289,\n",
       "        'We constructed a UC-VEGF-SMC three dimensional (3D) scaffold to explore its effect on blood vessel regeneration and bladder repair function in a rabbit model with bladder injury. Rabbit adipose tissue-derived stem cells (ADSCs) were cultured to construct pluripotent stem cell systems that can be induced to differentiate into urothelial cells (UCs) and smooth muscle cells (SMCs). Reverse transcriptase-polymerase chain reaction (RT-PCR), enzyme-linked immunosorbent assay (ELISA), immunofluorescence, MTT assay and HE staining were used in our study. Rabbit models were divided into an experimental group, control group and sham group. The bladder histology, urodynamics, smooth muscle function, stent degradation rate, urothelial permeability and biomechanical determination of rabbits were detected after grafting the scaffold. Correct stem cells based on the ADSC surface marker and found that CD90 and CD105 were positive and that CD34 and CD45 were negative. RT-PCR showed that ADSC-iPS cells expressed the marker gene of embryonic stem cells (ESCs), which indicated that Sox2, Klf4, Oct4 and c-Myc were inserted into the iPS nucleus and that the ADSC-iPS system was constructed successfully. Immunofluorescence and MTT assays indicated that iPS differentiated into mature SMCs and UCs. ELISA and HEMC culturing methods revealed that vascular endothelia growth factor (VEGF) could promote the growth of HMECs. Rabbit bladder repair function (urodynamics, smooth muscle function, urothelial permeability and biomechanical determination) was stronger in the experimental group than in the control group. UC-VEGF-SMC 3D nano scaffold exhibits good repair function for bladder damage, which may helpful for treatment of damaged bladders.']],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_test[['PMID', 'TEXT']].to_numpy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "cc5d31cc",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "All PyTorch model weights were used when initializing TFBertModel.\n",
      "\n",
      "All the weights of TFBertModel were initialized from the PyTorch model.\n",
      "If your task is similar to the task the model of the checkpoint was trained on, you can already use TFBertModel for predictions without further training.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'input_ids': <tf.Tensor: shape=(71, 256), dtype=int32, numpy=\n",
       " array([[  101,   151,  8816, ...,  1954,  3189,   102],\n",
       "        [  101,  1109,  1675, ...,   185,  7111,   102],\n",
       "        [  101,  1109,  2773, ...,  3382,  1233,   102],\n",
       "        ...,\n",
       "        [  101,  2695,   153, ...,   157,   119,   102],\n",
       "        [  101,  2082,  3551, ...,  1830, 26503,   102],\n",
       "        [  101,  1284,  3033, ...,  1115,  5844,   102]], dtype=int32)>}"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "{'in_silico': array([[1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.]], dtype=float32),\n",
       " 'organs': array([[1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.]], dtype=float32),\n",
       " 'other': array([[1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.]], dtype=float32),\n",
       " 'human': array([[1., 0.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.]], dtype=float32),\n",
       " 'in_vivo': array([[1., 0.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.]], dtype=float32),\n",
       " 'invertebrate': array([[1., 0.],\n",
       "        [1., 0.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.]], dtype=float32),\n",
       " 'primary_cells': array([[1., 0.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.]], dtype=float32),\n",
       " 'immortal_cell_line': array([[1., 0.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [0., 1.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.],\n",
       "        [1., 0.]], dtype=float32)}"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from transformers import TFBertModel, BertTokenizerFast, BertModel\n",
    "from tensorflow.keras.utils import to_categorical\n",
    "\n",
    "def setup_biobert():\n",
    "\t# Max length of tokens\n",
    "#\tmax_length = 128\n",
    "\tmax_length = 256\n",
    "\t# Load BioBERT tokenizer\n",
    "\ttokenizer = BertTokenizerFast.from_pretrained('dmis-lab/biobert-v1.1')\n",
    "\t# Load the Transformers BERT model\n",
    "\ttransformer_model = TFBertModel.from_pretrained('dmis-lab/biobert-v1.1', from_pt=True)\n",
    "\treturn transformer_model, transformer_model.config, max_length, tokenizer\n",
    "\n",
    "transformer_model, config, max_length, tokenizer = setup_biobert()\n",
    "\n",
    "def prepare_x_y_dev_test(data,max_length,tokenizer):\n",
    "\ttest_ys = {}\n",
    "\tfor label in all_labels:\n",
    "\t\ttest_y = to_categorical(data[label])\t\n",
    "\t\ttest_ys[label] = test_y\n",
    "\ttest_x = tokenizer(\n",
    "\t    text=data['TEXT'].to_list(),\n",
    "    \tadd_special_tokens=True,\n",
    "\t    max_length=max_length,\n",
    "\t    truncation=True,\n",
    "\t    padding=True, \n",
    "\t    return_tensors='tf',\n",
    "\t    return_token_type_ids = False,\n",
    "\t    return_attention_mask = False,\n",
    "\t    verbose = True)\n",
    "\tx={'input_ids': test_x['input_ids']}\n",
    "\ty = {}\n",
    "\tfor label in all_labels:\n",
    "\t\ty[label] = test_ys[label]\n",
    "\treturn x, y\n",
    "\n",
    "x, y = prepare_x_y_dev_test(data_test, max_length, tokenizer)\n",
    "\n",
    "display(x)\n",
    "display(y)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "e082c53c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:absl:At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3/3 [==============================] - 6s 1s/step\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'in_silico': array([[-1.2821354 ,  1.5166415 ],\n",
       "        [ 3.4218984 , -3.0335653 ],\n",
       "        [ 2.2962332 , -2.1601543 ],\n",
       "        [ 3.0511847 , -2.9212089 ],\n",
       "        [ 2.7034705 , -2.6561596 ],\n",
       "        [ 3.0328457 , -2.7379212 ],\n",
       "        [ 2.977708  , -2.869934  ],\n",
       "        [ 3.1903777 , -2.926043  ],\n",
       "        [ 2.8473723 , -2.6833549 ],\n",
       "        [ 3.0378098 , -2.7981951 ],\n",
       "        [ 3.2752156 , -3.1302323 ],\n",
       "        [ 3.3260486 , -3.0978427 ],\n",
       "        [ 1.9684745 , -1.8327421 ],\n",
       "        [ 3.0562363 , -2.7842572 ],\n",
       "        [ 3.3054936 , -3.0800784 ],\n",
       "        [ 3.0698197 , -2.8746367 ],\n",
       "        [ 3.15983   , -2.9699674 ],\n",
       "        [ 3.4055595 , -3.2769353 ],\n",
       "        [ 3.228362  , -2.8728113 ],\n",
       "        [ 2.1720924 , -2.0164433 ],\n",
       "        [ 2.5650463 , -2.4978714 ],\n",
       "        [ 2.7791328 , -2.623496  ],\n",
       "        [ 2.073139  , -2.0612695 ],\n",
       "        [ 3.5107555 , -3.2870429 ],\n",
       "        [ 2.4913619 , -2.377628  ],\n",
       "        [ 3.1075323 , -3.05359   ],\n",
       "        [ 3.3169994 , -3.0570366 ],\n",
       "        [ 3.7847989 , -3.15527   ],\n",
       "        [ 2.886548  , -2.5670774 ],\n",
       "        [ 2.6365743 , -2.3759255 ],\n",
       "        [ 2.928351  , -2.4713368 ],\n",
       "        [ 3.8245912 , -3.6365924 ],\n",
       "        [ 3.4416978 , -3.2666917 ],\n",
       "        [ 2.7794456 , -2.5858855 ],\n",
       "        [ 3.408824  , -3.137392  ],\n",
       "        [ 2.9783185 , -2.8553393 ],\n",
       "        [ 0.41535208, -0.2863528 ],\n",
       "        [ 2.57624   , -2.6237826 ],\n",
       "        [ 3.29727   , -2.9343386 ],\n",
       "        [ 3.024551  , -2.9791126 ],\n",
       "        [ 2.8868594 , -2.6543581 ],\n",
       "        [ 3.0076413 , -2.469382  ],\n",
       "        [ 3.2061205 , -3.0996406 ],\n",
       "        [ 2.3327405 , -2.1435962 ],\n",
       "        [ 2.1645691 , -1.9935067 ],\n",
       "        [ 3.105299  , -2.9372816 ],\n",
       "        [ 3.4500217 , -3.0753775 ],\n",
       "        [ 3.383481  , -3.3572423 ],\n",
       "        [ 3.3613222 , -3.0616972 ],\n",
       "        [ 2.8809996 , -2.7112267 ],\n",
       "        [-2.090922  ,  2.2693858 ],\n",
       "        [ 2.4372756 , -2.1413789 ],\n",
       "        [ 3.929412  , -3.8491821 ],\n",
       "        [ 0.54201806, -0.35639346],\n",
       "        [ 3.2258353 , -3.1285698 ],\n",
       "        [ 3.2983751 , -3.1168406 ],\n",
       "        [ 2.6714451 , -2.5634243 ],\n",
       "        [ 3.210947  , -2.989778  ],\n",
       "        [ 2.9051836 , -2.6284075 ],\n",
       "        [ 1.8241205 , -1.8557471 ],\n",
       "        [ 3.5071406 , -3.334969  ],\n",
       "        [ 3.36982   , -3.0900316 ],\n",
       "        [ 2.7536836 , -2.7442951 ],\n",
       "        [ 3.1181622 , -2.977318  ],\n",
       "        [ 3.2133749 , -3.0068376 ],\n",
       "        [ 3.68951   , -3.4713619 ],\n",
       "        [ 3.152788  , -3.1292686 ],\n",
       "        [ 3.5146456 , -3.6171916 ],\n",
       "        [ 1.5700527 , -1.565106  ],\n",
       "        [ 2.5486603 , -2.808735  ],\n",
       "        [ 3.3998141 , -3.2061114 ]], dtype=float32),\n",
       " 'organs': array([[ 3.2817514, -3.9947019],\n",
       "        [ 3.7402453, -4.5171947],\n",
       "        [ 3.7784996, -4.596871 ],\n",
       "        [ 3.3247156, -3.930563 ],\n",
       "        [ 3.5656724, -4.259259 ],\n",
       "        [ 3.5992665, -4.162931 ],\n",
       "        [ 3.9419782, -4.674676 ],\n",
       "        [ 3.4808109, -4.3541603],\n",
       "        [ 3.2929528, -4.0209317],\n",
       "        [ 3.511058 , -4.1621776],\n",
       "        [ 3.3591845, -4.0368752],\n",
       "        [ 3.5105898, -4.0746846],\n",
       "        [ 3.7346282, -4.2840223],\n",
       "        [ 3.9288986, -4.713434 ],\n",
       "        [ 3.5857723, -4.6313567],\n",
       "        [ 3.5802844, -4.1525373],\n",
       "        [ 3.7626753, -4.4797873],\n",
       "        [ 3.4164422, -4.1057715],\n",
       "        [ 3.7434146, -4.4969797],\n",
       "        [ 3.4969704, -4.1127253],\n",
       "        [ 3.7731328, -4.3795733],\n",
       "        [ 3.35518  , -4.0078664],\n",
       "        [ 3.755314 , -4.4123755],\n",
       "        [ 3.805778 , -4.398267 ],\n",
       "        [ 3.7551937, -4.378525 ],\n",
       "        [ 3.245247 , -3.9250631],\n",
       "        [ 3.1437907, -3.8088446],\n",
       "        [ 3.4531286, -4.2315326],\n",
       "        [ 3.3390803, -4.143837 ],\n",
       "        [ 3.7218628, -4.4607067],\n",
       "        [ 3.7167027, -4.4312773],\n",
       "        [ 3.6111503, -4.2332225],\n",
       "        [ 3.035886 , -3.5925162],\n",
       "        [ 3.5010345, -4.262104 ],\n",
       "        [ 3.7168736, -4.2765636],\n",
       "        [ 3.6508017, -4.1989164],\n",
       "        [ 2.8673697, -3.4709165],\n",
       "        [ 3.8771214, -4.5083528],\n",
       "        [ 3.4077985, -4.065859 ],\n",
       "        [ 3.6376433, -4.0836964],\n",
       "        [ 3.3727777, -3.842392 ],\n",
       "        [ 3.9139519, -4.804533 ],\n",
       "        [ 3.7816563, -4.3372765],\n",
       "        [ 3.6135976, -4.023019 ],\n",
       "        [ 3.1726556, -3.9406207],\n",
       "        [ 3.6954794, -4.177237 ],\n",
       "        [ 3.6667726, -4.307245 ],\n",
       "        [ 3.856974 , -4.5251427],\n",
       "        [ 3.551244 , -3.967998 ],\n",
       "        [ 3.5091925, -4.172511 ],\n",
       "        [ 2.501193 , -3.045411 ],\n",
       "        [ 3.6500468, -4.542844 ],\n",
       "        [ 2.7911665, -3.525805 ],\n",
       "        [ 3.7555382, -4.5586295],\n",
       "        [ 3.4126253, -4.1318703],\n",
       "        [ 3.346347 , -3.9647303],\n",
       "        [ 3.7332807, -4.5217733],\n",
       "        [ 3.5683775, -4.0599766],\n",
       "        [ 3.4510717, -4.1771617],\n",
       "        [ 3.2032578, -3.8867948],\n",
       "        [ 3.6707807, -4.2850723],\n",
       "        [ 3.7033448, -4.318704 ],\n",
       "        [ 3.776351 , -4.4433804],\n",
       "        [ 3.6986942, -4.255026 ],\n",
       "        [ 3.6974223, -4.180005 ],\n",
       "        [ 3.4751692, -3.981648 ],\n",
       "        [ 3.636255 , -4.09391  ],\n",
       "        [ 3.5496883, -3.8891764],\n",
       "        [ 3.3543658, -4.438706 ],\n",
       "        [ 3.5755847, -4.236013 ],\n",
       "        [ 3.5166206, -4.1331344]], dtype=float32),\n",
       " 'other': array([[-0.40166456,  0.51367706],\n",
       "        [ 0.9621093 , -0.8810735 ],\n",
       "        [-0.9454257 ,  1.0854125 ],\n",
       "        [ 2.1679337 , -2.0984814 ],\n",
       "        [ 0.19298613, -0.21317904],\n",
       "        [ 1.0040649 , -0.92011327],\n",
       "        [ 0.78762686, -0.65817237],\n",
       "        [ 0.24809653, -0.33194467],\n",
       "        [ 1.0257813 , -1.0443966 ],\n",
       "        [-0.24041733,  0.10273116],\n",
       "        [ 0.99484336, -1.155576  ],\n",
       "        [ 1.3207355 , -1.3224301 ],\n",
       "        [ 1.6798624 , -1.3437887 ],\n",
       "        [ 0.9187876 , -0.8163583 ],\n",
       "        [-0.06451681, -0.07398488],\n",
       "        [ 2.554493  , -2.3496633 ],\n",
       "        [ 1.1742619 , -1.1204058 ],\n",
       "        [ 0.68646264, -0.7013904 ],\n",
       "        [ 0.09381992, -0.05479985],\n",
       "        [ 0.5749947 , -0.5461613 ],\n",
       "        [ 0.42997462, -0.23792097],\n",
       "        [ 0.34505165, -0.35190013],\n",
       "        [-0.5399011 ,  0.47669113],\n",
       "        [ 1.5447942 , -1.3919687 ],\n",
       "        [ 0.9138516 , -0.8144499 ],\n",
       "        [ 0.8560536 , -1.0329057 ],\n",
       "        [ 1.2049776 , -1.3424479 ],\n",
       "        [ 1.2855288 , -1.2853428 ],\n",
       "        [-0.6449565 ,  0.36504823],\n",
       "        [ 1.1553801 , -1.0471975 ],\n",
       "        [ 0.6931713 , -0.6328045 ],\n",
       "        [ 1.1075875 , -1.1197858 ],\n",
       "        [ 1.7926602 , -1.987362  ],\n",
       "        [ 0.4101601 , -0.5233595 ],\n",
       "        [ 2.1527903 , -1.7560493 ],\n",
       "        [ 1.1972585 , -1.2791878 ],\n",
       "        [ 0.09661862, -0.23264734],\n",
       "        [ 0.03794735,  0.04174499],\n",
       "        [ 1.6693591 , -1.6032954 ],\n",
       "        [ 1.0344381 , -0.88437176],\n",
       "        [ 1.9904908 , -1.731878  ],\n",
       "        [ 2.1568046 , -1.5739949 ],\n",
       "        [ 0.6077487 , -0.75606436],\n",
       "        [ 1.1336107 , -0.939496  ],\n",
       "        [ 0.27302   , -0.16717897],\n",
       "        [ 1.7479824 , -1.5284636 ],\n",
       "        [ 1.1404185 , -0.99486846],\n",
       "        [ 0.50554127, -0.18601152],\n",
       "        [ 0.7947591 , -0.6408105 ],\n",
       "        [ 0.78486335, -0.8394949 ],\n",
       "        [ 0.2956047 , -0.09541147],\n",
       "        [-0.9436288 ,  0.879208  ],\n",
       "        [ 0.03718489, -0.24500957],\n",
       "        [-0.9249024 ,  1.1301783 ],\n",
       "        [ 0.70903724, -0.6906019 ],\n",
       "        [ 1.9959592 , -1.8381333 ],\n",
       "        [-0.5400891 ,  0.4342482 ],\n",
       "        [ 0.8942523 , -1.1143172 ],\n",
       "        [ 0.46667296, -0.352368  ],\n",
       "        [-1.2377222 ,  1.2955219 ],\n",
       "        [ 0.78287756, -0.83085567],\n",
       "        [ 0.7735487 , -0.9217486 ],\n",
       "        [ 0.341488  , -0.3555889 ],\n",
       "        [ 0.13502535, -0.16213949],\n",
       "        [ 1.3165636 , -1.3008578 ],\n",
       "        [ 1.3682356 , -1.3231322 ],\n",
       "        [ 0.06049009,  0.11583742],\n",
       "        [ 2.1266522 , -1.9390826 ],\n",
       "        [-1.747178  ,  1.7466393 ],\n",
       "        [-0.5190092 ,  0.6268129 ],\n",
       "        [ 0.55029744, -0.67973626]], dtype=float32),\n",
       " 'human': array([[ 2.6370425 , -2.8841686 ],\n",
       "        [ 2.009736  , -2.3570132 ],\n",
       "        [ 2.671741  , -2.9381902 ],\n",
       "        [ 3.0064514 , -2.5696807 ],\n",
       "        [ 2.1885815 , -2.1537483 ],\n",
       "        [ 2.156621  , -2.033045  ],\n",
       "        [ 2.8308954 , -2.8906364 ],\n",
       "        [ 1.1074141 , -1.3388402 ],\n",
       "        [ 2.254111  , -2.0984833 ],\n",
       "        [ 2.2629707 , -2.7234426 ],\n",
       "        [ 2.1127427 , -1.8512079 ],\n",
       "        [ 2.6087558 , -2.5748124 ],\n",
       "        [ 1.836996  , -1.368356  ],\n",
       "        [ 1.7394828 , -1.6412606 ],\n",
       "        [ 1.8619835 , -1.8989285 ],\n",
       "        [ 3.3797028 , -2.6163864 ],\n",
       "        [ 2.1557567 , -2.124765  ],\n",
       "        [ 2.289548  , -2.3570654 ],\n",
       "        [ 2.3544655 , -2.7492673 ],\n",
       "        [ 2.2560952 , -2.2818558 ],\n",
       "        [ 2.7045705 , -2.8112395 ],\n",
       "        [ 2.2955008 , -2.304322  ],\n",
       "        [ 2.7241523 , -2.883679  ],\n",
       "        [ 2.3964248 , -2.169784  ],\n",
       "        [ 2.849927  , -2.7299545 ],\n",
       "        [ 1.7870306 , -1.6706843 ],\n",
       "        [ 1.9288967 , -1.8020875 ],\n",
       "        [ 0.07806575, -0.12670247],\n",
       "        [ 2.2589333 , -2.7075324 ],\n",
       "        [ 1.7433424 , -1.553866  ],\n",
       "        [ 1.1628095 , -1.2050271 ],\n",
       "        [ 2.2158346 , -2.2326584 ],\n",
       "        [ 1.7909268 , -1.4148668 ],\n",
       "        [ 2.4951437 , -2.4290233 ],\n",
       "        [ 2.6602569 , -2.1351047 ],\n",
       "        [ 1.8812603 , -1.6005833 ],\n",
       "        [ 2.2185965 , -2.314104  ],\n",
       "        [ 3.096677  , -3.21019   ],\n",
       "        [ 1.5890037 , -1.2741545 ],\n",
       "        [ 2.9496102 , -2.8001738 ],\n",
       "        [ 2.149622  , -1.7052438 ],\n",
       "        [ 1.5536103 , -0.92498183],\n",
       "        [ 2.1325169 , -2.046735  ],\n",
       "        [ 2.3276467 , -1.9355065 ],\n",
       "        [ 1.6235354 , -1.7164202 ],\n",
       "        [ 2.0621657 , -1.5498385 ],\n",
       "        [ 2.1703775 , -2.0539868 ],\n",
       "        [ 3.0175087 , -3.152294  ],\n",
       "        [ 2.4712405 , -2.5225067 ],\n",
       "        [ 1.7432573 , -1.756391  ],\n",
       "        [ 2.582246  , -2.4841156 ],\n",
       "        [ 2.333936  , -2.6462307 ],\n",
       "        [ 2.7690315 , -3.106759  ],\n",
       "        [ 2.3890724 , -2.5857327 ],\n",
       "        [ 2.950757  , -3.106143  ],\n",
       "        [ 2.6296468 , -2.249942  ],\n",
       "        [ 2.2632883 , -2.6752274 ],\n",
       "        [ 1.3958743 , -1.1706762 ],\n",
       "        [ 2.4257774 , -2.3635323 ],\n",
       "        [ 3.2623067 , -3.3619387 ],\n",
       "        [ 2.1964855 , -2.3423834 ],\n",
       "        [ 1.4853045 , -1.44731   ],\n",
       "        [ 2.7727523 , -2.9613972 ],\n",
       "        [ 2.1427484 , -2.3481083 ],\n",
       "        [ 2.057242  , -1.8630967 ],\n",
       "        [ 1.895294  , -1.6732708 ],\n",
       "        [ 2.691469  , -2.522571  ],\n",
       "        [ 2.5073414 , -2.2586288 ],\n",
       "        [ 3.1313505 , -3.4621234 ],\n",
       "        [ 3.0944865 , -3.125587  ],\n",
       "        [ 1.8635579 , -1.8641605 ]], dtype=float32),\n",
       " 'in_vivo': array([[ 1.51852   , -1.8671274 ],\n",
       "        [ 2.3545434 , -2.8302627 ],\n",
       "        [ 2.4150028 , -2.8727853 ],\n",
       "        [-0.35090485, -0.30757305],\n",
       "        [ 1.5197313 , -2.2463078 ],\n",
       "        [ 0.85783035, -1.517188  ],\n",
       "        [ 1.1706192 , -1.797244  ],\n",
       "        [ 2.786101  , -3.4387276 ],\n",
       "        [ 0.9124561 , -1.7637348 ],\n",
       "        [ 2.4353447 , -3.0804286 ],\n",
       "        [ 0.9847651 , -1.7250049 ],\n",
       "        [ 0.9941143 , -1.7440016 ],\n",
       "        [-0.10147391, -0.72602975],\n",
       "        [ 1.3075715 , -2.1951232 ],\n",
       "        [ 1.7467995 , -2.4349597 ],\n",
       "        [-1.6477737 ,  0.98342437],\n",
       "        [ 1.1820561 , -1.9101044 ],\n",
       "        [ 1.4712847 , -2.1866102 ],\n",
       "        [ 1.9434928 , -2.7772572 ],\n",
       "        [ 1.5099701 , -2.171514  ],\n",
       "        [ 0.5597005 , -1.310336  ],\n",
       "        [ 1.5044407 , -2.0720549 ],\n",
       "        [ 1.4331651 , -2.129581  ],\n",
       "        [ 0.7051843 , -1.4253784 ],\n",
       "        [ 0.56727034, -1.330044  ],\n",
       "        [ 1.3912557 , -2.1468766 ],\n",
       "        [ 1.2860187 , -2.0164325 ],\n",
       "        [ 2.9289901 , -3.2970898 ],\n",
       "        [ 1.7674153 , -2.5136478 ],\n",
       "        [ 0.7787319 , -1.5797483 ],\n",
       "        [ 1.9449996 , -2.528733  ],\n",
       "        [ 1.0054766 , -1.6941906 ],\n",
       "        [ 1.1760432 , -1.8187026 ],\n",
       "        [ 1.2537378 , -1.9444965 ],\n",
       "        [ 0.19652043, -0.67186975],\n",
       "        [ 0.75651056, -1.4747181 ],\n",
       "        [ 1.6845845 , -1.9466046 ],\n",
       "        [ 1.2764146 , -1.9975458 ],\n",
       "        [ 0.74858207, -1.3273059 ],\n",
       "        [ 0.39427814, -0.9649074 ],\n",
       "        [-0.24524765, -0.49953625],\n",
       "        [-0.41350505, -0.3166562 ],\n",
       "        [ 0.7627532 , -1.4998872 ],\n",
       "        [-0.06978621, -0.5821424 ],\n",
       "        [ 1.6350491 , -2.0946572 ],\n",
       "        [-0.72828454, -0.00686105],\n",
       "        [ 1.4519628 , -2.118808  ],\n",
       "        [ 1.6287982 , -2.1489353 ],\n",
       "        [ 1.0646889 , -1.5257416 ],\n",
       "        [ 1.535983  , -2.2359605 ],\n",
       "        [ 2.2064269 , -2.1631637 ],\n",
       "        [ 1.533451  , -2.0805738 ],\n",
       "        [ 2.1049147 , -2.609148  ],\n",
       "        [ 1.3995192 , -2.080325  ],\n",
       "        [ 1.998339  , -2.6177921 ],\n",
       "        [-0.43107393, -0.32308775],\n",
       "        [ 2.348118  , -3.0110967 ],\n",
       "        [ 1.4166783 , -2.0088491 ],\n",
       "        [ 1.2086385 , -1.8419231 ],\n",
       "        [ 1.3678702 , -1.6682693 ],\n",
       "        [ 1.6318382 , -2.3643627 ],\n",
       "        [ 1.437105  , -2.083226  ],\n",
       "        [ 1.9234188 , -2.5471034 ],\n",
       "        [ 1.4588903 , -2.1793973 ],\n",
       "        [ 0.9447224 , -1.546924  ],\n",
       "        [ 1.0811268 , -1.6751591 ],\n",
       "        [-0.35500845, -0.17300236],\n",
       "        [ 0.94527596, -1.1445458 ],\n",
       "        [ 1.4018781 , -2.1180432 ],\n",
       "        [ 0.9320331 , -1.3901443 ],\n",
       "        [ 1.4507412 , -2.2124662 ]], dtype=float32),\n",
       " 'invertebrate': array([[ 1.9246646 , -1.7909763 ],\n",
       "        [ 2.4766166 , -2.667548  ],\n",
       "        [ 2.1792324 , -2.397158  ],\n",
       "        [ 2.728744  , -3.0361357 ],\n",
       "        [ 2.2805617 , -2.5214138 ],\n",
       "        [ 2.9444633 , -3.1643999 ],\n",
       "        [ 2.126928  , -2.2772126 ],\n",
       "        [ 2.435577  , -2.7555602 ],\n",
       "        [ 3.1183963 , -3.4127073 ],\n",
       "        [ 3.0554445 , -3.2429736 ],\n",
       "        [ 2.6959827 , -2.9536297 ],\n",
       "        [ 2.563386  , -2.807005  ],\n",
       "        [ 2.1490188 , -2.476859  ],\n",
       "        [ 2.8956916 , -3.204297  ],\n",
       "        [ 2.5810747 , -2.9819546 ],\n",
       "        [ 2.368208  , -2.8257525 ],\n",
       "        [ 2.5360253 , -2.8118048 ],\n",
       "        [ 2.9325235 , -3.283583  ],\n",
       "        [ 2.3341827 , -2.6662402 ],\n",
       "        [ 3.1867738 , -3.2570267 ],\n",
       "        [ 2.4194117 , -2.8222325 ],\n",
       "        [ 2.6978378 , -2.9289622 ],\n",
       "        [ 2.3395023 , -2.5473664 ],\n",
       "        [ 2.4367356 , -2.7434921 ],\n",
       "        [ 2.5420718 , -2.8482049 ],\n",
       "        [ 3.0193992 , -3.2196574 ],\n",
       "        [ 3.4549732 , -3.5553246 ],\n",
       "        [ 2.1775088 , -2.4964228 ],\n",
       "        [ 2.8539984 , -3.0624635 ],\n",
       "        [ 3.1024418 , -3.3831353 ],\n",
       "        [ 3.3771915 , -3.712832  ],\n",
       "        [ 2.4230294 , -2.7006853 ],\n",
       "        [ 3.8778994 , -4.0613427 ],\n",
       "        [ 2.8072288 , -3.0890508 ],\n",
       "        [ 1.8874184 , -2.04702   ],\n",
       "        [ 2.7526138 , -3.0137653 ],\n",
       "        [ 3.1536472 , -3.3643084 ],\n",
       "        [ 1.2485305 , -1.3915931 ],\n",
       "        [ 2.7857132 , -3.1220393 ],\n",
       "        [ 1.8693116 , -2.1807923 ],\n",
       "        [ 2.2911139 , -2.6293528 ],\n",
       "        [ 1.9583023 , -2.4019198 ],\n",
       "        [ 2.5500937 , -2.8723104 ],\n",
       "        [ 2.611225  , -2.9318788 ],\n",
       "        [ 1.4597584 , -1.9578565 ],\n",
       "        [ 2.4837787 , -2.9099464 ],\n",
       "        [ 3.1745203 , -3.6011386 ],\n",
       "        [ 2.1827924 , -2.363625  ],\n",
       "        [ 2.6369278 , -3.0170023 ],\n",
       "        [ 2.859058  , -3.264146  ],\n",
       "        [ 3.1535056 , -3.219143  ],\n",
       "        [ 2.8537488 , -3.0424435 ],\n",
       "        [ 2.6797266 , -2.860435  ],\n",
       "        [ 2.4196517 , -2.535346  ],\n",
       "        [ 2.4737546 , -2.4700527 ],\n",
       "        [ 2.5924106 , -3.1082883 ],\n",
       "        [ 2.4299734 , -2.5164056 ],\n",
       "        [ 2.7317705 , -3.2306726 ],\n",
       "        [ 3.4186084 , -3.6968174 ],\n",
       "        [ 1.9806192 , -2.1132782 ],\n",
       "        [ 3.005979  , -3.2929287 ],\n",
       "        [ 2.8202841 , -3.1292758 ],\n",
       "        [ 2.1964202 , -2.2219033 ],\n",
       "        [ 2.7895162 , -3.0013168 ],\n",
       "        [ 2.5498543 , -2.8855512 ],\n",
       "        [ 3.003661  , -3.2199702 ],\n",
       "        [ 2.2181904 , -2.5968933 ],\n",
       "        [ 0.17135282, -0.45558494],\n",
       "        [ 2.029862  , -1.9591022 ],\n",
       "        [ 1.9662331 , -2.121625  ],\n",
       "        [ 2.9545953 , -3.2624648 ]], dtype=float32),\n",
       " 'primary_cells': array([[ 3.3507347, -4.1040955],\n",
       "        [ 4.291791 , -5.430664 ],\n",
       "        [ 4.117705 , -5.3537827],\n",
       "        [ 4.114792 , -5.2947245],\n",
       "        [ 4.2921433, -5.3958044],\n",
       "        [ 4.1390414, -5.3069158],\n",
       "        [ 4.1253233, -5.52316  ],\n",
       "        [ 4.3929596, -5.432656 ],\n",
       "        [ 4.252099 , -5.4233613],\n",
       "        [ 4.2071977, -5.309999 ],\n",
       "        [ 4.166075 , -5.4005084],\n",
       "        [ 4.386834 , -5.6255984],\n",
       "        [ 3.8080435, -4.8412113],\n",
       "        [ 4.0437045, -5.1144023],\n",
       "        [ 4.409092 , -5.5113125],\n",
       "        [ 3.7074804, -4.855686 ],\n",
       "        [ 4.1827908, -5.374386 ],\n",
       "        [ 4.226941 , -5.4080467],\n",
       "        [ 4.1983624, -5.3514338],\n",
       "        [ 4.073249 , -5.128127 ],\n",
       "        [ 4.217262 , -5.4680347],\n",
       "        [ 4.276405 , -5.4972873],\n",
       "        [ 4.1494856, -5.329966 ],\n",
       "        [ 4.143772 , -5.2879324],\n",
       "        [ 3.9851038, -5.197293 ],\n",
       "        [ 4.200398 , -5.283137 ],\n",
       "        [ 3.9433882, -5.092589 ],\n",
       "        [ 3.970783 , -5.0229573],\n",
       "        [ 4.400037 , -5.6944385],\n",
       "        [ 4.1667747, -5.2337184],\n",
       "        [ 3.881637 , -4.775671 ],\n",
       "        [ 4.2821217, -5.580442 ],\n",
       "        [ 3.7883034, -4.7378135],\n",
       "        [ 4.2436724, -5.4769545],\n",
       "        [ 4.047121 , -5.1329403],\n",
       "        [ 3.899716 , -4.9788938],\n",
       "        [ 3.7762213, -4.5836864],\n",
       "        [ 4.1031847, -5.520472 ],\n",
       "        [ 4.189873 , -5.1162953],\n",
       "        [ 4.219598 , -5.4077344],\n",
       "        [ 4.1526723, -5.109006 ],\n",
       "        [ 3.694874 , -4.594301 ],\n",
       "        [ 4.177939 , -5.4178605],\n",
       "        [ 4.0504656, -5.0260606],\n",
       "        [ 4.685396 , -5.530649 ],\n",
       "        [ 3.913594 , -4.8520155],\n",
       "        [ 4.1411033, -5.112624 ],\n",
       "        [ 4.2861943, -5.545301 ],\n",
       "        [ 4.2351203, -5.4323893],\n",
       "        [ 4.2890787, -5.386998 ],\n",
       "        [ 3.303326 , -3.7567637],\n",
       "        [ 3.8034902, -4.8237395],\n",
       "        [ 4.1682086, -5.4099793],\n",
       "        [ 3.8384957, -4.9011955],\n",
       "        [ 4.1273108, -5.4661837],\n",
       "        [ 4.183442 , -5.379823 ],\n",
       "        [ 4.021262 , -5.228459 ],\n",
       "        [ 4.1048226, -5.1618176],\n",
       "        [ 4.0563517, -5.1146283],\n",
       "        [ 4.1216197, -5.1839423],\n",
       "        [ 4.1973715, -5.4630847],\n",
       "        [ 4.317311 , -5.4901404],\n",
       "        [ 4.1879883, -5.5035295],\n",
       "        [ 4.3373833, -5.422934 ],\n",
       "        [ 4.3461275, -5.477087 ],\n",
       "        [ 3.972229 , -5.008059 ],\n",
       "        [ 3.915841 , -5.106566 ],\n",
       "        [ 4.1608224, -5.2063584],\n",
       "        [ 3.685634 , -4.9698114],\n",
       "        [ 4.3324113, -5.469519 ],\n",
       "        [ 4.184461 , -5.326372 ]], dtype=float32),\n",
       " 'immortal_cell_line': array([[ 3.5499384, -3.9346633],\n",
       "        [ 4.5691752, -4.355023 ],\n",
       "        [ 4.0183086, -4.08289  ],\n",
       "        [ 4.926147 , -4.355493 ],\n",
       "        [ 4.572308 , -4.2077975],\n",
       "        [ 4.449206 , -4.012952 ],\n",
       "        [ 4.7604647, -4.4938326],\n",
       "        [ 4.2564664, -4.0565543],\n",
       "        [ 4.589191 , -4.15875  ],\n",
       "        [ 4.209842 , -4.3445477],\n",
       "        [ 4.496845 , -3.9188411],\n",
       "        [ 4.705251 , -4.1604857],\n",
       "        [ 4.901458 , -4.0561156],\n",
       "        [ 4.5407825, -4.2107515],\n",
       "        [ 4.3247857, -3.896208 ],\n",
       "        [ 5.0308223, -4.3749204],\n",
       "        [ 4.6615396, -4.2006783],\n",
       "        [ 4.372641 , -4.1302357],\n",
       "        [ 4.1768746, -4.162726 ],\n",
       "        [ 4.4551487, -4.155781 ],\n",
       "        [ 4.8122926, -4.363392 ],\n",
       "        [ 4.459082 , -4.0430493],\n",
       "        [ 4.18269  , -4.0592384],\n",
       "        [ 4.9761543, -4.3886547],\n",
       "        [ 4.826555 , -4.331989 ],\n",
       "        [ 4.289075 , -3.8781927],\n",
       "        [ 4.432732 , -4.165743 ],\n",
       "        [ 4.710972 , -4.4245224],\n",
       "        [ 4.136991 , -4.235283 ],\n",
       "        [ 4.715547 , -4.207591 ],\n",
       "        [ 4.3291473, -4.1523814],\n",
       "        [ 4.7636657, -4.2569537],\n",
       "        [ 4.1380973, -3.7551923],\n",
       "        [ 4.3691907, -3.9681814],\n",
       "        [ 5.1964617, -4.323548 ],\n",
       "        [ 4.452961 , -3.9405434],\n",
       "        [ 3.8419187, -3.975628 ],\n",
       "        [ 4.6021724, -4.0934315],\n",
       "        [ 4.6874847, -4.0364647],\n",
       "        [ 4.948539 , -4.336087 ],\n",
       "        [ 5.018401 , -4.3809466],\n",
       "        [ 5.020847 , -4.1428504],\n",
       "        [ 4.5688586, -4.1669884],\n",
       "        [ 4.7516904, -4.029565 ],\n",
       "        [ 4.430784 , -4.0297604],\n",
       "        [ 4.9346957, -4.2081   ],\n",
       "        [ 4.524227 , -4.4617767],\n",
       "        [ 4.617737 , -4.3151636],\n",
       "        [ 4.7443404, -4.400397 ],\n",
       "        [ 4.4374766, -4.0536323],\n",
       "        [ 2.9411967, -3.4816973],\n",
       "        [ 3.9499328, -3.9124262],\n",
       "        [ 4.3453507, -4.2223663],\n",
       "        [ 3.7477055, -3.948093 ],\n",
       "        [ 4.980707 , -4.5163317],\n",
       "        [ 4.936404 , -4.274944 ],\n",
       "        [ 4.4812727, -4.3141828],\n",
       "        [ 4.362141 , -4.1023917],\n",
       "        [ 4.4542494, -4.1459875],\n",
       "        [ 4.1520257, -3.992587 ],\n",
       "        [ 4.657829 , -4.3752923],\n",
       "        [ 4.385536 , -3.9596245],\n",
       "        [ 4.5113187, -4.142894 ],\n",
       "        [ 4.527628 , -4.226538 ],\n",
       "        [ 4.622207 , -4.1302423],\n",
       "        [ 4.4034996, -3.9831212],\n",
       "        [ 4.4553437, -3.7832115],\n",
       "        [ 5.0030265, -4.3268237],\n",
       "        [ 3.7834227, -3.7995195],\n",
       "        [ 4.3875656, -4.0412555],\n",
       "        [ 4.5009637, -4.1601   ]], dtype=float32)}"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from tensorflow.keras.models import Model, load_model\n",
    "\n",
    "model_path = project_root / \"models/goldhamster/goldhamster_model.h5\"\n",
    "\n",
    "\n",
    "# Load model\n",
    "model = load_model(model_path)\n",
    "# Run predictions\n",
    "predictions = model.predict(x)\n",
    "predictions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "2a19684f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def print_predictions(predictions, train_dev_test_dir, test_file, out_file, skipped_test):\n",
    "\t#print(len(predictions[label]))\t\n",
    "\twith open(os.path.join(train_dev_test_dir,out_file), \"w\") as writer:\n",
    "\t\twith open(os.path.join(train_dev_test_dir,test_file), \"r\") as reader:\n",
    "\t\t\tlines = reader.readlines()\n",
    "\t\t\tdoc_index = 0\n",
    "\t\t\tfor line in lines:\n",
    "\t\t\t\tpmid, str_labels = line.strip().split(\"\\t\")\n",
    "\t\t\t\tlist_labels = []\n",
    "\t\t\t\tif doc_index not in skipped_test:\n",
    "\t\t\t\t\tfor label in predictions:\n",
    "\t\t\t\t\t\tarr_pred = predictions[label][doc_index]\n",
    "\t\t\t\t\t\t#print(arr_pred)\n",
    "\t\t\t\t\t\tif arr_pred[1]>arr_pred[0]:\n",
    "\t\t\t\t\t\t\tlist_labels.append(label)\n",
    "\t\t\t\t\tdoc_index += 1\n",
    "\t\t\t\t#print(pmid,list_labels)\n",
    "\t\t\t\twriter.write(pmid+\"\\t\"+','.join(list_labels)+\"\\n\")\n",
    "\t\twriter.close()\n",
    "\n",
    "print_predictions(predictions, labels_dir, \"test0.txt\", 'preds_0.txt', skipped_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "72f38d01",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0f861f67",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "f1e7bd71",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "All PyTorch model weights were used when initializing TFBertModel.\n",
      "\n",
      "All the weights of TFBertModel were initialized from the PyTorch model.\n",
      "If your task is similar to the task the model of the checkpoint was trained on, you can already use TFBertModel for predictions without further training.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Model: \"BERT_MultiLabel_MultiClass\"\n",
      "__________________________________________________________________________________________________\n",
      " Layer (type)                Output Shape                 Param #   Connected to                  \n",
      "==================================================================================================\n",
      " input_ids (InputLayer)      [(None, 256)]                0         []                            \n",
      "                                                                                                  \n",
      " bert (TFBertMainLayer)      TFBaseModelOutputWithPooli   1083102   ['input_ids[0][0]']           \n",
      "                             ngAndCrossAttentions(last_   72                                      \n",
      "                             hidden_state=(None, 256, 7                                           \n",
      "                             68),                                                                 \n",
      "                              pooler_output=(None, 768)                                           \n",
      "                             , past_key_values=None, hi                                           \n",
      "                             dden_states=None, attentio                                           \n",
      "                             ns=None, cross_attentions=                                           \n",
      "                             None)                                                                \n",
      "                                                                                                  \n",
      " pooled_output (Dropout)     (None, 768)                  0         ['bert[0][1]']                \n",
      "                                                                                                  \n",
      " human (Dense)               (None, 2)                    1538      ['pooled_output[0][0]']       \n",
      "                                                                                                  \n",
      " immortal_cell_line (Dense)  (None, 2)                    1538      ['pooled_output[0][0]']       \n",
      "                                                                                                  \n",
      " in_silico (Dense)           (None, 2)                    1538      ['pooled_output[0][0]']       \n",
      "                                                                                                  \n",
      " in_vivo (Dense)             (None, 2)                    1538      ['pooled_output[0][0]']       \n",
      "                                                                                                  \n",
      " invertebrate (Dense)        (None, 2)                    1538      ['pooled_output[0][0]']       \n",
      "                                                                                                  \n",
      " organs (Dense)              (None, 2)                    1538      ['pooled_output[0][0]']       \n",
      "                                                                                                  \n",
      " other (Dense)               (None, 2)                    1538      ['pooled_output[0][0]']       \n",
      "                                                                                                  \n",
      " primary_cells (Dense)       (None, 2)                    1538      ['pooled_output[0][0]']       \n",
      "                                                                                                  \n",
      "==================================================================================================\n",
      "Total params: 108322576 (413.22 MB)\n",
      "Trainable params: 108322576 (413.22 MB)\n",
      "Non-trainable params: 0 (0.00 Byte)\n",
      "__________________________________________________________________________________________________\n"
     ]
    },
    {
     "ename": "TypeError",
     "evalue": "TextEncodeInput must be Union[TextInputSequence, Tuple[InputSequence, InputSequence]]",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mTypeError\u001b[39m                                 Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[21]\u001b[39m\u001b[32m, line 104\u001b[39m\n\u001b[32m    102\u001b[39m split = \u001b[32m1\u001b[39m\n\u001b[32m    103\u001b[39m name = \u001b[33m\"\u001b[39m\u001b[33mgoldhamster\u001b[39m\u001b[33m\"\u001b[39m\n\u001b[32m--> \u001b[39m\u001b[32m104\u001b[39m \u001b[43mtrain_bert_goldhamster2\u001b[49m\u001b[43m(\u001b[49m\u001b[43mpapers_dir\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mlabels_dir\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43mtrain\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m+\u001b[49m\u001b[38;5;28;43mstr\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43msplit\u001b[49m\u001b[43m)\u001b[49m\u001b[43m+\u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43m.txt\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43mdev\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m+\u001b[49m\u001b[38;5;28;43mstr\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43msplit\u001b[49m\u001b[43m)\u001b[49m\u001b[43m+\u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43m.txt\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[32m    105\u001b[39m predict_with_model(papers_dir, labels_dir, \u001b[33m'\u001b[39m\u001b[33mtest\u001b[39m\u001b[33m'\u001b[39m+\u001b[38;5;28mstr\u001b[39m(split)+\u001b[33m'\u001b[39m\u001b[33m.txt\u001b[39m\u001b[33m'\u001b[39m, \u001b[33m'\u001b[39m\u001b[33mpreds_\u001b[39m\u001b[33m'\u001b[39m+\u001b[38;5;28mstr\u001b[39m(split)+\u001b[33m'\u001b[39m\u001b[33m_\u001b[39m\u001b[33m'\u001b[39m+name+\u001b[33m'\u001b[39m\u001b[33m.txt\u001b[39m\u001b[33m'\u001b[39m)\n",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[21]\u001b[39m\u001b[32m, line 84\u001b[39m, in \u001b[36mtrain_bert_goldhamster2\u001b[39m\u001b[34m(docs_dir, train_dev_test_dir, train_file, dev_file)\u001b[39m\n\u001b[32m     82\u001b[39m model = build_model(transformer_model,config,max_length,data_train)\n\u001b[32m     83\u001b[39m \u001b[38;5;66;03m# train model\u001b[39;00m\n\u001b[32m---> \u001b[39m\u001b[32m84\u001b[39m \u001b[43mtrain_model\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmodel\u001b[49m\u001b[43m,\u001b[49m\u001b[43mdata_train\u001b[49m\u001b[43m,\u001b[49m\u001b[43mdata_dev\u001b[49m\u001b[43m,\u001b[49m\u001b[43mmax_length\u001b[49m\u001b[43m,\u001b[49m\u001b[43mtokenizer\u001b[49m\u001b[43m,\u001b[49m\u001b[43mlearning_rate\u001b[49m\u001b[43m,\u001b[49m\u001b[43mbatch_size\u001b[49m\u001b[43m,\u001b[49m\u001b[43mepochs\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[21]\u001b[39m\u001b[32m, line 63\u001b[39m, in \u001b[36mtrain_model\u001b[39m\u001b[34m(model, data_train, data_dev, max_length, tokenizer, learning_rate, batch_size, epochs)\u001b[39m\n\u001b[32m     58\u001b[39m model.compile(\n\u001b[32m     59\u001b[39m     optimizer = optimizer,\n\u001b[32m     60\u001b[39m     loss = loss, \n\u001b[32m     61\u001b[39m     metrics = metric)\n\u001b[32m     62\u001b[39m \u001b[38;5;66;03m# trainig data\u001b[39;00m\n\u001b[32m---> \u001b[39m\u001b[32m63\u001b[39m x_train, y_train = \u001b[43mprepare_x_y_dev_test\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdata_train\u001b[49m\u001b[43m,\u001b[49m\u001b[43mmax_length\u001b[49m\u001b[43m,\u001b[49m\u001b[43mtokenizer\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m     64\u001b[39m \u001b[38;5;66;03m# validation data\u001b[39;00m\n\u001b[32m     65\u001b[39m x_val, y_val = prepare_x_y_dev_test(data_dev,max_length,tokenizer)\n",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[17]\u001b[39m\u001b[32m, line 21\u001b[39m, in \u001b[36mprepare_x_y_dev_test\u001b[39m\u001b[34m(data, max_length, tokenizer)\u001b[39m\n\u001b[32m     19\u001b[39m \t\ttest_y = to_categorical(data[label])\t\n\u001b[32m     20\u001b[39m \t\ttest_ys[label] = test_y\n\u001b[32m---> \u001b[39m\u001b[32m21\u001b[39m \ttest_x = \u001b[43mtokenizer\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m     22\u001b[39m \u001b[43m\t    \u001b[49m\u001b[43mtext\u001b[49m\u001b[43m=\u001b[49m\u001b[43mdata\u001b[49m\u001b[43m[\u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43mTEXT\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m.\u001b[49m\u001b[43mto_list\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m     23\u001b[39m \u001b[43m    \t\u001b[49m\u001b[43madd_special_tokens\u001b[49m\u001b[43m=\u001b[49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[32m     24\u001b[39m \u001b[43m\t    \u001b[49m\u001b[43mmax_length\u001b[49m\u001b[43m=\u001b[49m\u001b[43mmax_length\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m     25\u001b[39m \u001b[43m\t    \u001b[49m\u001b[43mtruncation\u001b[49m\u001b[43m=\u001b[49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[32m     26\u001b[39m \u001b[43m\t    \u001b[49m\u001b[43mpadding\u001b[49m\u001b[43m=\u001b[49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[32m     27\u001b[39m \u001b[43m\t    \u001b[49m\u001b[43mreturn_tensors\u001b[49m\u001b[43m=\u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43mtf\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[32m     28\u001b[39m \u001b[43m\t    \u001b[49m\u001b[43mreturn_token_type_ids\u001b[49m\u001b[43m \u001b[49m\u001b[43m=\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[32m     29\u001b[39m \u001b[43m\t    \u001b[49m\u001b[43mreturn_attention_mask\u001b[49m\u001b[43m \u001b[49m\u001b[43m=\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[32m     30\u001b[39m \u001b[43m\t    \u001b[49m\u001b[43mverbose\u001b[49m\u001b[43m \u001b[49m\u001b[43m=\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m)\u001b[49m\n\u001b[32m     31\u001b[39m \tx={\u001b[33m'\u001b[39m\u001b[33minput_ids\u001b[39m\u001b[33m'\u001b[39m: test_x[\u001b[33m'\u001b[39m\u001b[33minput_ids\u001b[39m\u001b[33m'\u001b[39m]}\n\u001b[32m     32\u001b[39m \ty = {}\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniforge3/envs/3r-monitoring/lib/python3.11/site-packages/transformers/tokenization_utils_base.py:2887\u001b[39m, in \u001b[36mPreTrainedTokenizerBase.__call__\u001b[39m\u001b[34m(self, text, text_pair, text_target, text_pair_target, add_special_tokens, padding, truncation, max_length, stride, is_split_into_words, pad_to_multiple_of, padding_side, return_tensors, return_token_type_ids, return_attention_mask, return_overflowing_tokens, return_special_tokens_mask, return_offsets_mapping, return_length, verbose, **kwargs)\u001b[39m\n\u001b[32m   2885\u001b[39m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mself\u001b[39m._in_target_context_manager:\n\u001b[32m   2886\u001b[39m         \u001b[38;5;28mself\u001b[39m._switch_to_input_mode()\n\u001b[32m-> \u001b[39m\u001b[32m2887\u001b[39m     encodings = \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_call_one\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtext\u001b[49m\u001b[43m=\u001b[49m\u001b[43mtext\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtext_pair\u001b[49m\u001b[43m=\u001b[49m\u001b[43mtext_pair\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mall_kwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m   2888\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m text_target \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[32m   2889\u001b[39m     \u001b[38;5;28mself\u001b[39m._switch_to_target_mode()\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniforge3/envs/3r-monitoring/lib/python3.11/site-packages/transformers/tokenization_utils_base.py:2975\u001b[39m, in \u001b[36mPreTrainedTokenizerBase._call_one\u001b[39m\u001b[34m(self, text, text_pair, add_special_tokens, padding, truncation, max_length, stride, is_split_into_words, pad_to_multiple_of, padding_side, return_tensors, return_token_type_ids, return_attention_mask, return_overflowing_tokens, return_special_tokens_mask, return_offsets_mapping, return_length, verbose, split_special_tokens, **kwargs)\u001b[39m\n\u001b[32m   2970\u001b[39m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\n\u001b[32m   2971\u001b[39m             \u001b[33mf\u001b[39m\u001b[33m\"\u001b[39m\u001b[33mbatch length of `text`: \u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mlen\u001b[39m(text)\u001b[38;5;132;01m}\u001b[39;00m\u001b[33m does not match batch length of `text_pair`:\u001b[39m\u001b[33m\"\u001b[39m\n\u001b[32m   2972\u001b[39m             \u001b[33mf\u001b[39m\u001b[33m\"\u001b[39m\u001b[33m \u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mlen\u001b[39m(text_pair)\u001b[38;5;132;01m}\u001b[39;00m\u001b[33m.\u001b[39m\u001b[33m\"\u001b[39m\n\u001b[32m   2973\u001b[39m         )\n\u001b[32m   2974\u001b[39m     batch_text_or_text_pairs = \u001b[38;5;28mlist\u001b[39m(\u001b[38;5;28mzip\u001b[39m(text, text_pair)) \u001b[38;5;28;01mif\u001b[39;00m text_pair \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;28;01melse\u001b[39;00m text\n\u001b[32m-> \u001b[39m\u001b[32m2975\u001b[39m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43mbatch_encode_plus\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m   2976\u001b[39m \u001b[43m        \u001b[49m\u001b[43mbatch_text_or_text_pairs\u001b[49m\u001b[43m=\u001b[49m\u001b[43mbatch_text_or_text_pairs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   2977\u001b[39m \u001b[43m        \u001b[49m\u001b[43madd_special_tokens\u001b[49m\u001b[43m=\u001b[49m\u001b[43madd_special_tokens\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   2978\u001b[39m \u001b[43m        \u001b[49m\u001b[43mpadding\u001b[49m\u001b[43m=\u001b[49m\u001b[43mpadding\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   2979\u001b[39m \u001b[43m        \u001b[49m\u001b[43mtruncation\u001b[49m\u001b[43m=\u001b[49m\u001b[43mtruncation\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   2980\u001b[39m \u001b[43m        \u001b[49m\u001b[43mmax_length\u001b[49m\u001b[43m=\u001b[49m\u001b[43mmax_length\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   2981\u001b[39m \u001b[43m        \u001b[49m\u001b[43mstride\u001b[49m\u001b[43m=\u001b[49m\u001b[43mstride\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   2982\u001b[39m \u001b[43m        \u001b[49m\u001b[43mis_split_into_words\u001b[49m\u001b[43m=\u001b[49m\u001b[43mis_split_into_words\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   2983\u001b[39m \u001b[43m        \u001b[49m\u001b[43mpad_to_multiple_of\u001b[49m\u001b[43m=\u001b[49m\u001b[43mpad_to_multiple_of\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   2984\u001b[39m \u001b[43m        \u001b[49m\u001b[43mpadding_side\u001b[49m\u001b[43m=\u001b[49m\u001b[43mpadding_side\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   2985\u001b[39m \u001b[43m        \u001b[49m\u001b[43mreturn_tensors\u001b[49m\u001b[43m=\u001b[49m\u001b[43mreturn_tensors\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   2986\u001b[39m \u001b[43m        \u001b[49m\u001b[43mreturn_token_type_ids\u001b[49m\u001b[43m=\u001b[49m\u001b[43mreturn_token_type_ids\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   2987\u001b[39m \u001b[43m        \u001b[49m\u001b[43mreturn_attention_mask\u001b[49m\u001b[43m=\u001b[49m\u001b[43mreturn_attention_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   2988\u001b[39m \u001b[43m        \u001b[49m\u001b[43mreturn_overflowing_tokens\u001b[49m\u001b[43m=\u001b[49m\u001b[43mreturn_overflowing_tokens\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   2989\u001b[39m \u001b[43m        \u001b[49m\u001b[43mreturn_special_tokens_mask\u001b[49m\u001b[43m=\u001b[49m\u001b[43mreturn_special_tokens_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   2990\u001b[39m \u001b[43m        \u001b[49m\u001b[43mreturn_offsets_mapping\u001b[49m\u001b[43m=\u001b[49m\u001b[43mreturn_offsets_mapping\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   2991\u001b[39m \u001b[43m        \u001b[49m\u001b[43mreturn_length\u001b[49m\u001b[43m=\u001b[49m\u001b[43mreturn_length\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   2992\u001b[39m \u001b[43m        \u001b[49m\u001b[43mverbose\u001b[49m\u001b[43m=\u001b[49m\u001b[43mverbose\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   2993\u001b[39m \u001b[43m        \u001b[49m\u001b[43msplit_special_tokens\u001b[49m\u001b[43m=\u001b[49m\u001b[43msplit_special_tokens\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   2994\u001b[39m \u001b[43m        \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   2995\u001b[39m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m   2996\u001b[39m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[32m   2997\u001b[39m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m.encode_plus(\n\u001b[32m   2998\u001b[39m         text=text,\n\u001b[32m   2999\u001b[39m         text_pair=text_pair,\n\u001b[32m   (...)\u001b[39m\u001b[32m   3017\u001b[39m         **kwargs,\n\u001b[32m   3018\u001b[39m     )\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniforge3/envs/3r-monitoring/lib/python3.11/site-packages/transformers/tokenization_utils_base.py:3177\u001b[39m, in \u001b[36mPreTrainedTokenizerBase.batch_encode_plus\u001b[39m\u001b[34m(self, batch_text_or_text_pairs, add_special_tokens, padding, truncation, max_length, stride, is_split_into_words, pad_to_multiple_of, padding_side, return_tensors, return_token_type_ids, return_attention_mask, return_overflowing_tokens, return_special_tokens_mask, return_offsets_mapping, return_length, verbose, split_special_tokens, **kwargs)\u001b[39m\n\u001b[32m   3167\u001b[39m \u001b[38;5;66;03m# Backward compatibility for 'truncation_strategy', 'pad_to_max_length'\u001b[39;00m\n\u001b[32m   3168\u001b[39m padding_strategy, truncation_strategy, max_length, kwargs = \u001b[38;5;28mself\u001b[39m._get_padding_truncation_strategies(\n\u001b[32m   3169\u001b[39m     padding=padding,\n\u001b[32m   3170\u001b[39m     truncation=truncation,\n\u001b[32m   (...)\u001b[39m\u001b[32m   3174\u001b[39m     **kwargs,\n\u001b[32m   3175\u001b[39m )\n\u001b[32m-> \u001b[39m\u001b[32m3177\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_batch_encode_plus\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m   3178\u001b[39m \u001b[43m    \u001b[49m\u001b[43mbatch_text_or_text_pairs\u001b[49m\u001b[43m=\u001b[49m\u001b[43mbatch_text_or_text_pairs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3179\u001b[39m \u001b[43m    \u001b[49m\u001b[43madd_special_tokens\u001b[49m\u001b[43m=\u001b[49m\u001b[43madd_special_tokens\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3180\u001b[39m \u001b[43m    \u001b[49m\u001b[43mpadding_strategy\u001b[49m\u001b[43m=\u001b[49m\u001b[43mpadding_strategy\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3181\u001b[39m \u001b[43m    \u001b[49m\u001b[43mtruncation_strategy\u001b[49m\u001b[43m=\u001b[49m\u001b[43mtruncation_strategy\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3182\u001b[39m \u001b[43m    \u001b[49m\u001b[43mmax_length\u001b[49m\u001b[43m=\u001b[49m\u001b[43mmax_length\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3183\u001b[39m \u001b[43m    \u001b[49m\u001b[43mstride\u001b[49m\u001b[43m=\u001b[49m\u001b[43mstride\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3184\u001b[39m \u001b[43m    \u001b[49m\u001b[43mis_split_into_words\u001b[49m\u001b[43m=\u001b[49m\u001b[43mis_split_into_words\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3185\u001b[39m \u001b[43m    \u001b[49m\u001b[43mpad_to_multiple_of\u001b[49m\u001b[43m=\u001b[49m\u001b[43mpad_to_multiple_of\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3186\u001b[39m \u001b[43m    \u001b[49m\u001b[43mpadding_side\u001b[49m\u001b[43m=\u001b[49m\u001b[43mpadding_side\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3187\u001b[39m \u001b[43m    \u001b[49m\u001b[43mreturn_tensors\u001b[49m\u001b[43m=\u001b[49m\u001b[43mreturn_tensors\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3188\u001b[39m \u001b[43m    \u001b[49m\u001b[43mreturn_token_type_ids\u001b[49m\u001b[43m=\u001b[49m\u001b[43mreturn_token_type_ids\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3189\u001b[39m \u001b[43m    \u001b[49m\u001b[43mreturn_attention_mask\u001b[49m\u001b[43m=\u001b[49m\u001b[43mreturn_attention_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3190\u001b[39m \u001b[43m    \u001b[49m\u001b[43mreturn_overflowing_tokens\u001b[49m\u001b[43m=\u001b[49m\u001b[43mreturn_overflowing_tokens\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3191\u001b[39m \u001b[43m    \u001b[49m\u001b[43mreturn_special_tokens_mask\u001b[49m\u001b[43m=\u001b[49m\u001b[43mreturn_special_tokens_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3192\u001b[39m \u001b[43m    \u001b[49m\u001b[43mreturn_offsets_mapping\u001b[49m\u001b[43m=\u001b[49m\u001b[43mreturn_offsets_mapping\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3193\u001b[39m \u001b[43m    \u001b[49m\u001b[43mreturn_length\u001b[49m\u001b[43m=\u001b[49m\u001b[43mreturn_length\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3194\u001b[39m \u001b[43m    \u001b[49m\u001b[43mverbose\u001b[49m\u001b[43m=\u001b[49m\u001b[43mverbose\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3195\u001b[39m \u001b[43m    \u001b[49m\u001b[43msplit_special_tokens\u001b[49m\u001b[43m=\u001b[49m\u001b[43msplit_special_tokens\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3196\u001b[39m \u001b[43m    \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3197\u001b[39m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniforge3/envs/3r-monitoring/lib/python3.11/site-packages/transformers/tokenization_utils_fast.py:539\u001b[39m, in \u001b[36mPreTrainedTokenizerFast._batch_encode_plus\u001b[39m\u001b[34m(self, batch_text_or_text_pairs, add_special_tokens, padding_strategy, truncation_strategy, max_length, stride, is_split_into_words, pad_to_multiple_of, padding_side, return_tensors, return_token_type_ids, return_attention_mask, return_overflowing_tokens, return_special_tokens_mask, return_offsets_mapping, return_length, verbose, split_special_tokens)\u001b[39m\n\u001b[32m    536\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m._tokenizer.encode_special_tokens != split_special_tokens:\n\u001b[32m    537\u001b[39m     \u001b[38;5;28mself\u001b[39m._tokenizer.encode_special_tokens = split_special_tokens\n\u001b[32m--> \u001b[39m\u001b[32m539\u001b[39m encodings = \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_tokenizer\u001b[49m\u001b[43m.\u001b[49m\u001b[43mencode_batch\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    540\u001b[39m \u001b[43m    \u001b[49m\u001b[43mbatch_text_or_text_pairs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    541\u001b[39m \u001b[43m    \u001b[49m\u001b[43madd_special_tokens\u001b[49m\u001b[43m=\u001b[49m\u001b[43madd_special_tokens\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    542\u001b[39m \u001b[43m    \u001b[49m\u001b[43mis_pretokenized\u001b[49m\u001b[43m=\u001b[49m\u001b[43mis_split_into_words\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    543\u001b[39m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    545\u001b[39m \u001b[38;5;66;03m# Convert encoding to dict\u001b[39;00m\n\u001b[32m    546\u001b[39m \u001b[38;5;66;03m# `Tokens` has type: Tuple[\u001b[39;00m\n\u001b[32m    547\u001b[39m \u001b[38;5;66;03m#                       List[Dict[str, List[List[int]]]] or List[Dict[str, 2D-Tensor]],\u001b[39;00m\n\u001b[32m    548\u001b[39m \u001b[38;5;66;03m#                       List[EncodingFast]\u001b[39;00m\n\u001b[32m    549\u001b[39m \u001b[38;5;66;03m#                    ]\u001b[39;00m\n\u001b[32m    550\u001b[39m \u001b[38;5;66;03m# with nested dimensions corresponding to batch, overflows, sequence length\u001b[39;00m\n\u001b[32m    551\u001b[39m tokens_and_encodings = [\n\u001b[32m    552\u001b[39m     \u001b[38;5;28mself\u001b[39m._convert_encoding(\n\u001b[32m    553\u001b[39m         encoding=encoding,\n\u001b[32m   (...)\u001b[39m\u001b[32m    562\u001b[39m     \u001b[38;5;28;01mfor\u001b[39;00m encoding \u001b[38;5;129;01min\u001b[39;00m encodings\n\u001b[32m    563\u001b[39m ]\n",
      "\u001b[31mTypeError\u001b[39m: TextEncodeInput must be Union[TextInputSequence, Tuple[InputSequence, InputSequence]]"
     ]
    }
   ],
   "source": [
    "from tensorflow.keras.losses import CategoricalCrossentropy\n",
    "from tensorflow.keras.metrics import CategoricalAccuracy\n",
    "from tensorflow.keras.layers import Input, Dropout, Dense\n",
    "from tensorflow.keras.models import Model, load_model\n",
    "# from tensorflow.keras.optimizers import Adam\n",
    "from tensorflow.keras.optimizers.legacy import Adam\n",
    "from tensorflow.keras.initializers import TruncatedNormal\n",
    "\n",
    "\n",
    "save_model_path = project_root / \"models/goldhamster_custom/goldhamster_model.h5\"\n",
    "\n",
    "# hyperparameters\n",
    "learning_rate = 1e-04\n",
    "batch_size = 32\n",
    "epochs = 10\n",
    "\n",
    "#######################################\n",
    "### ------- Build the model ------- ###\n",
    "def build_model(transformer_model,config,max_length,data):\n",
    "\t# TF Keras documentation: https://www.tensorflow.org/api_docs/python/tf/keras/Model# Load the MainLayer\n",
    "\tbert = transformer_model.layers[0]\n",
    "\t# Build your model input\n",
    "\tinput_ids = Input(shape=(max_length,), name='input_ids', dtype='int32')\n",
    "\tinputs = {'input_ids': input_ids}\n",
    "\t# Load the Transformers BERT model as a layer in a Keras model\n",
    "\tbert_model = bert(inputs)[1]\n",
    "\tdropout = Dropout(config.hidden_dropout_prob, name='pooled_output')\n",
    "\tpooled_output = dropout(bert_model, training=False)\n",
    "\t# Then build your model output\n",
    "\t# one per output\n",
    "\toutputs = {}\n",
    "\tfor label in all_labels:\n",
    "\t\tlabel_output = Dense(units=len(data[label+'_label'].value_counts()), kernel_initializer=TruncatedNormal(stddev=config.initializer_range), name=label)(pooled_output)\n",
    "\t\toutputs[label] = label_output\n",
    "\t# And combine it all in a model object\n",
    "\tmodel = Model(inputs=inputs, outputs=outputs, name='BERT_MultiLabel_MultiClass')\n",
    "\t# Take a look at the model\n",
    "\tmodel.summary()\n",
    "\treturn model\n",
    "\n",
    "#######################################\n",
    "### ------- Train the model ------- ###\n",
    "def train_model(model,data_train,data_dev,max_length,tokenizer,learning_rate,batch_size,epochs):\n",
    "\t# Set an optimizer\n",
    "\toptimizer = Adam(\n",
    "    \tlearning_rate,\n",
    "    \tepsilon=1e-08,\n",
    "    \tdecay=0.01,\n",
    "    \tclipnorm=1.0)\n",
    "\t# Set loss and metrics\n",
    "\tloss = {}\n",
    "\tfor label in all_labels:\n",
    "\t\tloss[label] = CategoricalCrossentropy(from_logits = True)\n",
    "\tmetric = {}\n",
    "\tfor label in all_labels:\n",
    "\t\tmetric[label] = CategoricalAccuracy('accuracy')\n",
    "\t# Compile the model\n",
    "\tmodel.compile(\n",
    "\t    optimizer = optimizer,\n",
    "\t    loss = loss, \n",
    "\t    metrics = metric)\n",
    "\t# trainig data\n",
    "\tx_train, y_train = prepare_x_y_dev_test(data_train,max_length,tokenizer)\n",
    "\t# validation data\n",
    "\tx_val, y_val = prepare_x_y_dev_test(data_dev,max_length,tokenizer)\n",
    "\t# Fit the model\n",
    "\thistory = model.fit(\n",
    "\t\tx_train, y_train,\n",
    "\t\tvalidation_data=(x_val,y_val),\n",
    "\t    batch_size=batch_size,\n",
    "\t    epochs=epochs\n",
    "\t)\n",
    "\tmodel.save(model_path)\n",
    "\n",
    "def train_bert_goldhamster2(docs_dir,train_dev_test_dir,train_file,dev_file):\n",
    "\t# import data\n",
    "\tdata_train, tsv_file_train, skipped_train = pre_process_data(docs_dir,train_dev_test_dir,train_file)\n",
    "\tdata_dev, tsv_file_dev, skipped_dev = pre_process_data(docs_dir,train_dev_test_dir,dev_file)\n",
    "\t# set up model\n",
    "\ttransformer_model, config, max_length, tokenizer = setup_biobert() \n",
    "\t# build model\n",
    "\tmodel = build_model(transformer_model,config,max_length,data_train)\n",
    "\t# train model\n",
    "\ttrain_model(model,data_train,data_dev,max_length,tokenizer,learning_rate,batch_size,epochs)\n",
    "\n",
    "\n",
    "def predict_with_model(docs_dir,train_dev_test_dir,test_file,out_file):\n",
    "\t# import data\n",
    "\tdata_test, tsv_file_test, skipped_test = pre_process_data(docs_dir,train_dev_test_dir,test_file)\n",
    "\t# set up model\n",
    "\ttransformer_model, config, max_length, tokenizer = setup_biobert()\n",
    "\t# prepare test data\n",
    "\tx, y = prepare_x_y_dev_test(data_test,max_length,tokenizer)\n",
    "\t# Load model\n",
    "\tmodel = load_model('model.h5')\n",
    "\t# Run predictions\n",
    "\tpredictions = model.predict(x)\n",
    "\tprint(model.summary())\n",
    "\tprint_predictions(predictions,train_dev_test_dir,test_file,out_file,skipped_test)\n",
    "\t\n",
    "\n",
    "split = 1\n",
    "name = \"goldhamster\"\n",
    "train_bert_goldhamster2(papers_dir, labels_dir, 'train'+str(split)+'.txt', 'dev'+str(split)+'.txt')\n",
    "predict_with_model(papers_dir, labels_dir, 'test'+str(split)+'.txt', 'preds_'+str(split)+'_'+name+'.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fb95f62a",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "3r-monitoring",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
